// cyanmaple/CMUI/es/styleInit.es.js
function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

var _global = createCommonjsModule(function (module) {
// https://github.com/zloirock/core-js/issues/86#issuecomment-115759028
var global = module.exports = typeof window != 'undefined' && window.Math == Math
  ? window : typeof self != 'undefined' && self.Math == Math ? self
  // eslint-disable-next-line no-new-func
  : Function('return this')();
if (typeof __g == 'number') __g = global; // eslint-disable-line no-undef
});

var _core = createCommonjsModule(function (module) {
var core = module.exports = { version: '2.6.11' };
if (typeof __e == 'number') __e = core; // eslint-disable-line no-undef
});
var _core_1 = _core.version;

var _library = false;

var _shared = createCommonjsModule(function (module) {
var SHARED = '__core-js_shared__';
var store = _global[SHARED] || (_global[SHARED] = {});

(module.exports = function (key, value) {
  return store[key] || (store[key] = value !== undefined ? value : {});
})('versions', []).push({
  version: _core.version,
  mode:  'global',
  copyright: '© 2019 Denis Pushkarev (zloirock.ru)'
});
});

var id = 0;
var px = Math.random();
var _uid = function (key) {
  return 'Symbol('.concat(key === undefined ? '' : key, ')_', (++id + px).toString(36));
};

var _wks = createCommonjsModule(function (module) {
var store = _shared('wks');

var Symbol = _global.Symbol;
var USE_SYMBOL = typeof Symbol == 'function';

var $exports = module.exports = function (name) {
  return store[name] || (store[name] =
    USE_SYMBOL && Symbol[name] || (USE_SYMBOL ? Symbol : _uid)('Symbol.' + name));
};

$exports.store = store;
});

var f = _wks;

var _wksExt = {
	f: f
};

var _isObject = function (it) {
  return typeof it === 'object' ? it !== null : typeof it === 'function';
};

var _anObject = function (it) {
  if (!_isObject(it)) throw TypeError(it + ' is not an object!');
  return it;
};

var _fails = function (exec) {
  try {
    return !!exec();
  } catch (e) {
    return true;
  }
};

// Thank's IE8 for his funny defineProperty
var _descriptors = !_fails(function () {
  return Object.defineProperty({}, 'a', { get: function () { return 7; } }).a != 7;
});

var document$1 = _global.document;
// typeof document.createElement is 'object' in old IE
var is = _isObject(document$1) && _isObject(document$1.createElement);
var _domCreate = function (it) {
  return is ? document$1.createElement(it) : {};
};

var _ie8DomDefine = !_descriptors && !_fails(function () {
  return Object.defineProperty(_domCreate('div'), 'a', { get: function () { return 7; } }).a != 7;
});

// 7.1.1 ToPrimitive(input [, PreferredType])

// instead of the ES6 spec version, we didn't implement @@toPrimitive case
// and the second argument - flag - preferred type is a string
var _toPrimitive = function (it, S) {
  if (!_isObject(it)) return it;
  var fn, val;
  if (S && typeof (fn = it.toString) == 'function' && !_isObject(val = fn.call(it))) return val;
  if (typeof (fn = it.valueOf) == 'function' && !_isObject(val = fn.call(it))) return val;
  if (!S && typeof (fn = it.toString) == 'function' && !_isObject(val = fn.call(it))) return val;
  throw TypeError("Can't convert object to primitive value");
};

var dP = Object.defineProperty;

var f$1 = _descriptors ? Object.defineProperty : function defineProperty(O, P, Attributes) {
  _anObject(O);
  P = _toPrimitive(P, true);
  _anObject(Attributes);
  if (_ie8DomDefine) try {
    return dP(O, P, Attributes);
  } catch (e) { /* empty */ }
  if ('get' in Attributes || 'set' in Attributes) throw TypeError('Accessors not supported!');
  if ('value' in Attributes) O[P] = Attributes.value;
  return O;
};

var _objectDp = {
	f: f$1
};

var defineProperty = _objectDp.f;
var _wksDefine = function (name) {
  var $Symbol = _core.Symbol || (_core.Symbol =  _global.Symbol || {});
  if (name.charAt(0) != '_' && !(name in $Symbol)) defineProperty($Symbol, name, { value: _wksExt.f(name) });
};

_wksDefine('asyncIterator');

var hasOwnProperty = {}.hasOwnProperty;
var _has = function (it, key) {
  return hasOwnProperty.call(it, key);
};

var _propertyDesc = function (bitmap, value) {
  return {
    enumerable: !(bitmap & 1),
    configurable: !(bitmap & 2),
    writable: !(bitmap & 4),
    value: value
  };
};

var _hide = _descriptors ? function (object, key, value) {
  return _objectDp.f(object, key, _propertyDesc(1, value));
} : function (object, key, value) {
  object[key] = value;
  return object;
};

var _functionToString = _shared('native-function-to-string', Function.toString);

var _redefine = createCommonjsModule(function (module) {
var SRC = _uid('src');

var TO_STRING = 'toString';
var TPL = ('' + _functionToString).split(TO_STRING);

_core.inspectSource = function (it) {
  return _functionToString.call(it);
};

(module.exports = function (O, key, val, safe) {
  var isFunction = typeof val == 'function';
  if (isFunction) _has(val, 'name') || _hide(val, 'name', key);
  if (O[key] === val) return;
  if (isFunction) _has(val, SRC) || _hide(val, SRC, O[key] ? '' + O[key] : TPL.join(String(key)));
  if (O === _global) {
    O[key] = val;
  } else if (!safe) {
    delete O[key];
    _hide(O, key, val);
  } else if (O[key]) {
    O[key] = val;
  } else {
    _hide(O, key, val);
  }
// add fake Function#toString for correct work wrapped methods / constructors with methods like LoDash isNative
})(Function.prototype, TO_STRING, function toString() {
  return typeof this == 'function' && this[SRC] || _functionToString.call(this);
});
});

var _aFunction = function (it) {
  if (typeof it != 'function') throw TypeError(it + ' is not a function!');
  return it;
};

// optional / simple context binding

var _ctx = function (fn, that, length) {
  _aFunction(fn);
  if (that === undefined) return fn;
  switch (length) {
    case 1: return function (a) {
      return fn.call(that, a);
    };
    case 2: return function (a, b) {
      return fn.call(that, a, b);
    };
    case 3: return function (a, b, c) {
      return fn.call(that, a, b, c);
    };
  }
  return function (/* ...args */) {
    return fn.apply(that, arguments);
  };
};

var PROTOTYPE = 'prototype';

var $export = function (type, name, source) {
  var IS_FORCED = type & $export.F;
  var IS_GLOBAL = type & $export.G;
  var IS_STATIC = type & $export.S;
  var IS_PROTO = type & $export.P;
  var IS_BIND = type & $export.B;
  var target = IS_GLOBAL ? _global : IS_STATIC ? _global[name] || (_global[name] = {}) : (_global[name] || {})[PROTOTYPE];
  var exports = IS_GLOBAL ? _core : _core[name] || (_core[name] = {});
  var expProto = exports[PROTOTYPE] || (exports[PROTOTYPE] = {});
  var key, own, out, exp;
  if (IS_GLOBAL) source = name;
  for (key in source) {
    // contains in native
    own = !IS_FORCED && target && target[key] !== undefined;
    // export native or passed
    out = (own ? target : source)[key];
    // bind timers to global for call from export context
    exp = IS_BIND && own ? _ctx(out, _global) : IS_PROTO && typeof out == 'function' ? _ctx(Function.call, out) : out;
    // extend global
    if (target) _redefine(target, key, out, type & $export.U);
    // export
    if (exports[key] != out) _hide(exports, key, exp);
    if (IS_PROTO && expProto[key] != out) expProto[key] = out;
  }
};
_global.core = _core;
// type bitmap
$export.F = 1;   // forced
$export.G = 2;   // global
$export.S = 4;   // static
$export.P = 8;   // proto
$export.B = 16;  // bind
$export.W = 32;  // wrap
$export.U = 64;  // safe
$export.R = 128; // real proto method for `library`
var _export = $export;

var _meta = createCommonjsModule(function (module) {
var META = _uid('meta');


var setDesc = _objectDp.f;
var id = 0;
var isExtensible = Object.isExtensible || function () {
  return true;
};
var FREEZE = !_fails(function () {
  return isExtensible(Object.preventExtensions({}));
});
var setMeta = function (it) {
  setDesc(it, META, { value: {
    i: 'O' + ++id, // object ID
    w: {}          // weak collections IDs
  } });
};
var fastKey = function (it, create) {
  // return primitive with prefix
  if (!_isObject(it)) return typeof it == 'symbol' ? it : (typeof it == 'string' ? 'S' : 'P') + it;
  if (!_has(it, META)) {
    // can't set metadata to uncaught frozen object
    if (!isExtensible(it)) return 'F';
    // not necessary to add metadata
    if (!create) return 'E';
    // add missing metadata
    setMeta(it);
  // return object ID
  } return it[META].i;
};
var getWeak = function (it, create) {
  if (!_has(it, META)) {
    // can't set metadata to uncaught frozen object
    if (!isExtensible(it)) return true;
    // not necessary to add metadata
    if (!create) return false;
    // add missing metadata
    setMeta(it);
  // return hash weak collections IDs
  } return it[META].w;
};
// add metadata on freeze-family methods calling
var onFreeze = function (it) {
  if (FREEZE && meta.NEED && isExtensible(it) && !_has(it, META)) setMeta(it);
  return it;
};
var meta = module.exports = {
  KEY: META,
  NEED: false,
  fastKey: fastKey,
  getWeak: getWeak,
  onFreeze: onFreeze
};
});
var _meta_1 = _meta.KEY;
var _meta_2 = _meta.NEED;
var _meta_3 = _meta.fastKey;
var _meta_4 = _meta.getWeak;
var _meta_5 = _meta.onFreeze;

var def = _objectDp.f;

var TAG = _wks('toStringTag');

var _setToStringTag = function (it, tag, stat) {
  if (it && !_has(it = stat ? it : it.prototype, TAG)) def(it, TAG, { configurable: true, value: tag });
};

var toString = {}.toString;

var _cof = function (it) {
  return toString.call(it).slice(8, -1);
};

// fallback for non-array-like ES3 and non-enumerable old V8 strings

// eslint-disable-next-line no-prototype-builtins
var _iobject = Object('z').propertyIsEnumerable(0) ? Object : function (it) {
  return _cof(it) == 'String' ? it.split('') : Object(it);
};

// 7.2.1 RequireObjectCoercible(argument)
var _defined = function (it) {
  if (it == undefined) throw TypeError("Can't call method on  " + it);
  return it;
};

// to indexed object, toObject with fallback for non-array-like ES3 strings


var _toIobject = function (it) {
  return _iobject(_defined(it));
};

// 7.1.4 ToInteger
var ceil = Math.ceil;
var floor = Math.floor;
var _toInteger = function (it) {
  return isNaN(it = +it) ? 0 : (it > 0 ? floor : ceil)(it);
};

// 7.1.15 ToLength

var min = Math.min;
var _toLength = function (it) {
  return it > 0 ? min(_toInteger(it), 0x1fffffffffffff) : 0; // pow(2, 53) - 1 == 9007199254740991
};

var max = Math.max;
var min$1 = Math.min;
var _toAbsoluteIndex = function (index, length) {
  index = _toInteger(index);
  return index < 0 ? max(index + length, 0) : min$1(index, length);
};

// false -> Array#indexOf
// true  -> Array#includes



var _arrayIncludes = function (IS_INCLUDES) {
  return function ($this, el, fromIndex) {
    var O = _toIobject($this);
    var length = _toLength(O.length);
    var index = _toAbsoluteIndex(fromIndex, length);
    var value;
    // Array#includes uses SameValueZero equality algorithm
    // eslint-disable-next-line no-self-compare
    if (IS_INCLUDES && el != el) while (length > index) {
      value = O[index++];
      // eslint-disable-next-line no-self-compare
      if (value != value) return true;
    // Array#indexOf ignores holes, Array#includes - not
    } else for (;length > index; index++) if (IS_INCLUDES || index in O) {
      if (O[index] === el) return IS_INCLUDES || index || 0;
    } return !IS_INCLUDES && -1;
  };
};

var shared = _shared('keys');

var _sharedKey = function (key) {
  return shared[key] || (shared[key] = _uid(key));
};

var arrayIndexOf = _arrayIncludes(false);
var IE_PROTO = _sharedKey('IE_PROTO');

var _objectKeysInternal = function (object, names) {
  var O = _toIobject(object);
  var i = 0;
  var result = [];
  var key;
  for (key in O) if (key != IE_PROTO) _has(O, key) && result.push(key);
  // Don't enum bug & hidden keys
  while (names.length > i) if (_has(O, key = names[i++])) {
    ~arrayIndexOf(result, key) || result.push(key);
  }
  return result;
};

// IE 8- don't enum bug keys
var _enumBugKeys = (
  'constructor,hasOwnProperty,isPrototypeOf,propertyIsEnumerable,toLocaleString,toString,valueOf'
).split(',');

// 19.1.2.14 / 15.2.3.14 Object.keys(O)



var _objectKeys = Object.keys || function keys(O) {
  return _objectKeysInternal(O, _enumBugKeys);
};

var f$2 = Object.getOwnPropertySymbols;

var _objectGops = {
	f: f$2
};

var f$3 = {}.propertyIsEnumerable;

var _objectPie = {
	f: f$3
};

// all enumerable object keys, includes symbols



var _enumKeys = function (it) {
  var result = _objectKeys(it);
  var getSymbols = _objectGops.f;
  if (getSymbols) {
    var symbols = getSymbols(it);
    var isEnum = _objectPie.f;
    var i = 0;
    var key;
    while (symbols.length > i) if (isEnum.call(it, key = symbols[i++])) result.push(key);
  } return result;
};

// 7.2.2 IsArray(argument)

var _isArray = Array.isArray || function isArray(arg) {
  return _cof(arg) == 'Array';
};

// 7.1.13 ToObject(argument)

var _toObject = function (it) {
  return Object(_defined(it));
};

var _objectDps = _descriptors ? Object.defineProperties : function defineProperties(O, Properties) {
  _anObject(O);
  var keys = _objectKeys(Properties);
  var length = keys.length;
  var i = 0;
  var P;
  while (length > i) _objectDp.f(O, P = keys[i++], Properties[P]);
  return O;
};

var document$2 = _global.document;
var _html = document$2 && document$2.documentElement;

// 19.1.2.2 / 15.2.3.5 Object.create(O [, Properties])



var IE_PROTO$1 = _sharedKey('IE_PROTO');
var Empty = function () { /* empty */ };
var PROTOTYPE$1 = 'prototype';

// Create object with fake `null` prototype: use iframe Object with cleared prototype
var createDict = function () {
  // Thrash, waste and sodomy: IE GC bug
  var iframe = _domCreate('iframe');
  var i = _enumBugKeys.length;
  var lt = '<';
  var gt = '>';
  var iframeDocument;
  iframe.style.display = 'none';
  _html.appendChild(iframe);
  iframe.src = 'javascript:'; // eslint-disable-line no-script-url
  // createDict = iframe.contentWindow.Object;
  // html.removeChild(iframe);
  iframeDocument = iframe.contentWindow.document;
  iframeDocument.open();
  iframeDocument.write(lt + 'script' + gt + 'document.F=Object' + lt + '/script' + gt);
  iframeDocument.close();
  createDict = iframeDocument.F;
  while (i--) delete createDict[PROTOTYPE$1][_enumBugKeys[i]];
  return createDict();
};

var _objectCreate = Object.create || function create(O, Properties) {
  var result;
  if (O !== null) {
    Empty[PROTOTYPE$1] = _anObject(O);
    result = new Empty();
    Empty[PROTOTYPE$1] = null;
    // add "__proto__" for Object.getPrototypeOf polyfill
    result[IE_PROTO$1] = O;
  } else result = createDict();
  return Properties === undefined ? result : _objectDps(result, Properties);
};

// 19.1.2.7 / 15.2.3.4 Object.getOwnPropertyNames(O)

var hiddenKeys = _enumBugKeys.concat('length', 'prototype');

var f$4 = Object.getOwnPropertyNames || function getOwnPropertyNames(O) {
  return _objectKeysInternal(O, hiddenKeys);
};

var _objectGopn = {
	f: f$4
};

// fallback for IE11 buggy Object.getOwnPropertyNames with iframe and window

var gOPN = _objectGopn.f;
var toString$1 = {}.toString;

var windowNames = typeof window == 'object' && window && Object.getOwnPropertyNames
  ? Object.getOwnPropertyNames(window) : [];

var getWindowNames = function (it) {
  try {
    return gOPN(it);
  } catch (e) {
    return windowNames.slice();
  }
};

var f$5 = function getOwnPropertyNames(it) {
  return windowNames && toString$1.call(it) == '[object Window]' ? getWindowNames(it) : gOPN(_toIobject(it));
};

var _objectGopnExt = {
	f: f$5
};

var gOPD = Object.getOwnPropertyDescriptor;

var f$6 = _descriptors ? gOPD : function getOwnPropertyDescriptor(O, P) {
  O = _toIobject(O);
  P = _toPrimitive(P, true);
  if (_ie8DomDefine) try {
    return gOPD(O, P);
  } catch (e) { /* empty */ }
  if (_has(O, P)) return _propertyDesc(!_objectPie.f.call(O, P), O[P]);
};

var _objectGopd = {
	f: f$6
};

// ECMAScript 6 symbols shim





var META = _meta.KEY;





















var gOPD$1 = _objectGopd.f;
var dP$1 = _objectDp.f;
var gOPN$1 = _objectGopnExt.f;
var $Symbol = _global.Symbol;
var $JSON = _global.JSON;
var _stringify = $JSON && $JSON.stringify;
var PROTOTYPE$2 = 'prototype';
var HIDDEN = _wks('_hidden');
var TO_PRIMITIVE = _wks('toPrimitive');
var isEnum = {}.propertyIsEnumerable;
var SymbolRegistry = _shared('symbol-registry');
var AllSymbols = _shared('symbols');
var OPSymbols = _shared('op-symbols');
var ObjectProto = Object[PROTOTYPE$2];
var USE_NATIVE = typeof $Symbol == 'function' && !!_objectGops.f;
var QObject = _global.QObject;
// Don't use setters in Qt Script, https://github.com/zloirock/core-js/issues/173
var setter = !QObject || !QObject[PROTOTYPE$2] || !QObject[PROTOTYPE$2].findChild;

// fallback for old Android, https://code.google.com/p/v8/issues/detail?id=687
var setSymbolDesc = _descriptors && _fails(function () {
  return _objectCreate(dP$1({}, 'a', {
    get: function () { return dP$1(this, 'a', { value: 7 }).a; }
  })).a != 7;
}) ? function (it, key, D) {
  var protoDesc = gOPD$1(ObjectProto, key);
  if (protoDesc) delete ObjectProto[key];
  dP$1(it, key, D);
  if (protoDesc && it !== ObjectProto) dP$1(ObjectProto, key, protoDesc);
} : dP$1;

var wrap = function (tag) {
  var sym = AllSymbols[tag] = _objectCreate($Symbol[PROTOTYPE$2]);
  sym._k = tag;
  return sym;
};

var isSymbol = USE_NATIVE && typeof $Symbol.iterator == 'symbol' ? function (it) {
  return typeof it == 'symbol';
} : function (it) {
  return it instanceof $Symbol;
};

var $defineProperty = function defineProperty(it, key, D) {
  if (it === ObjectProto) $defineProperty(OPSymbols, key, D);
  _anObject(it);
  key = _toPrimitive(key, true);
  _anObject(D);
  if (_has(AllSymbols, key)) {
    if (!D.enumerable) {
      if (!_has(it, HIDDEN)) dP$1(it, HIDDEN, _propertyDesc(1, {}));
      it[HIDDEN][key] = true;
    } else {
      if (_has(it, HIDDEN) && it[HIDDEN][key]) it[HIDDEN][key] = false;
      D = _objectCreate(D, { enumerable: _propertyDesc(0, false) });
    } return setSymbolDesc(it, key, D);
  } return dP$1(it, key, D);
};
var $defineProperties = function defineProperties(it, P) {
  _anObject(it);
  var keys = _enumKeys(P = _toIobject(P));
  var i = 0;
  var l = keys.length;
  var key;
  while (l > i) $defineProperty(it, key = keys[i++], P[key]);
  return it;
};
var $create = function create(it, P) {
  return P === undefined ? _objectCreate(it) : $defineProperties(_objectCreate(it), P);
};
var $propertyIsEnumerable = function propertyIsEnumerable(key) {
  var E = isEnum.call(this, key = _toPrimitive(key, true));
  if (this === ObjectProto && _has(AllSymbols, key) && !_has(OPSymbols, key)) return false;
  return E || !_has(this, key) || !_has(AllSymbols, key) || _has(this, HIDDEN) && this[HIDDEN][key] ? E : true;
};
var $getOwnPropertyDescriptor = function getOwnPropertyDescriptor(it, key) {
  it = _toIobject(it);
  key = _toPrimitive(key, true);
  if (it === ObjectProto && _has(AllSymbols, key) && !_has(OPSymbols, key)) return;
  var D = gOPD$1(it, key);
  if (D && _has(AllSymbols, key) && !(_has(it, HIDDEN) && it[HIDDEN][key])) D.enumerable = true;
  return D;
};
var $getOwnPropertyNames = function getOwnPropertyNames(it) {
  var names = gOPN$1(_toIobject(it));
  var result = [];
  var i = 0;
  var key;
  while (names.length > i) {
    if (!_has(AllSymbols, key = names[i++]) && key != HIDDEN && key != META) result.push(key);
  } return result;
};
var $getOwnPropertySymbols = function getOwnPropertySymbols(it) {
  var IS_OP = it === ObjectProto;
  var names = gOPN$1(IS_OP ? OPSymbols : _toIobject(it));
  var result = [];
  var i = 0;
  var key;
  while (names.length > i) {
    if (_has(AllSymbols, key = names[i++]) && (IS_OP ? _has(ObjectProto, key) : true)) result.push(AllSymbols[key]);
  } return result;
};

// 19.4.1.1 Symbol([description])
if (!USE_NATIVE) {
  $Symbol = function Symbol() {
    if (this instanceof $Symbol) throw TypeError('Symbol is not a constructor!');
    var tag = _uid(arguments.length > 0 ? arguments[0] : undefined);
    var $set = function (value) {
      if (this === ObjectProto) $set.call(OPSymbols, value);
      if (_has(this, HIDDEN) && _has(this[HIDDEN], tag)) this[HIDDEN][tag] = false;
      setSymbolDesc(this, tag, _propertyDesc(1, value));
    };
    if (_descriptors && setter) setSymbolDesc(ObjectProto, tag, { configurable: true, set: $set });
    return wrap(tag);
  };
  _redefine($Symbol[PROTOTYPE$2], 'toString', function toString() {
    return this._k;
  });

  _objectGopd.f = $getOwnPropertyDescriptor;
  _objectDp.f = $defineProperty;
  _objectGopn.f = _objectGopnExt.f = $getOwnPropertyNames;
  _objectPie.f = $propertyIsEnumerable;
  _objectGops.f = $getOwnPropertySymbols;

  if (_descriptors && !_library) {
    _redefine(ObjectProto, 'propertyIsEnumerable', $propertyIsEnumerable, true);
  }

  _wksExt.f = function (name) {
    return wrap(_wks(name));
  };
}

_export(_export.G + _export.W + _export.F * !USE_NATIVE, { Symbol: $Symbol });

for (var es6Symbols = (
  // 19.4.2.2, 19.4.2.3, 19.4.2.4, 19.4.2.6, 19.4.2.8, 19.4.2.9, 19.4.2.10, 19.4.2.11, 19.4.2.12, 19.4.2.13, 19.4.2.14
  'hasInstance,isConcatSpreadable,iterator,match,replace,search,species,split,toPrimitive,toStringTag,unscopables'
).split(','), j = 0; es6Symbols.length > j;)_wks(es6Symbols[j++]);

for (var wellKnownSymbols = _objectKeys(_wks.store), k = 0; wellKnownSymbols.length > k;) _wksDefine(wellKnownSymbols[k++]);

_export(_export.S + _export.F * !USE_NATIVE, 'Symbol', {
  // 19.4.2.1 Symbol.for(key)
  'for': function (key) {
    return _has(SymbolRegistry, key += '')
      ? SymbolRegistry[key]
      : SymbolRegistry[key] = $Symbol(key);
  },
  // 19.4.2.5 Symbol.keyFor(sym)
  keyFor: function keyFor(sym) {
    if (!isSymbol(sym)) throw TypeError(sym + ' is not a symbol!');
    for (var key in SymbolRegistry) if (SymbolRegistry[key] === sym) return key;
  },
  useSetter: function () { setter = true; },
  useSimple: function () { setter = false; }
});

_export(_export.S + _export.F * !USE_NATIVE, 'Object', {
  // 19.1.2.2 Object.create(O [, Properties])
  create: $create,
  // 19.1.2.4 Object.defineProperty(O, P, Attributes)
  defineProperty: $defineProperty,
  // 19.1.2.3 Object.defineProperties(O, Properties)
  defineProperties: $defineProperties,
  // 19.1.2.6 Object.getOwnPropertyDescriptor(O, P)
  getOwnPropertyDescriptor: $getOwnPropertyDescriptor,
  // 19.1.2.7 Object.getOwnPropertyNames(O)
  getOwnPropertyNames: $getOwnPropertyNames,
  // 19.1.2.8 Object.getOwnPropertySymbols(O)
  getOwnPropertySymbols: $getOwnPropertySymbols
});

// Chrome 38 and 39 `Object.getOwnPropertySymbols` fails on primitives
// https://bugs.chromium.org/p/v8/issues/detail?id=3443
var FAILS_ON_PRIMITIVES = _fails(function () { _objectGops.f(1); });

_export(_export.S + _export.F * FAILS_ON_PRIMITIVES, 'Object', {
  getOwnPropertySymbols: function getOwnPropertySymbols(it) {
    return _objectGops.f(_toObject(it));
  }
});

// 24.3.2 JSON.stringify(value [, replacer [, space]])
$JSON && _export(_export.S + _export.F * (!USE_NATIVE || _fails(function () {
  var S = $Symbol();
  // MS Edge converts symbol values to JSON as {}
  // WebKit converts symbol values to JSON as null
  // V8 throws on boxed symbols
  return _stringify([S]) != '[null]' || _stringify({ a: S }) != '{}' || _stringify(Object(S)) != '{}';
})), 'JSON', {
  stringify: function stringify(it) {
    var args = [it];
    var i = 1;
    var replacer, $replacer;
    while (arguments.length > i) args.push(arguments[i++]);
    $replacer = replacer = args[1];
    if (!_isObject(replacer) && it === undefined || isSymbol(it)) return; // IE8 returns string on undefined
    if (!_isArray(replacer)) replacer = function (key, value) {
      if (typeof $replacer == 'function') value = $replacer.call(this, key, value);
      if (!isSymbol(value)) return value;
    };
    args[1] = replacer;
    return _stringify.apply($JSON, args);
  }
});

// 19.4.3.4 Symbol.prototype[@@toPrimitive](hint)
$Symbol[PROTOTYPE$2][TO_PRIMITIVE] || _hide($Symbol[PROTOTYPE$2], TO_PRIMITIVE, $Symbol[PROTOTYPE$2].valueOf);
// 19.4.3.5 Symbol.prototype[@@toStringTag]
_setToStringTag($Symbol, 'Symbol');
// 20.2.1.9 Math[@@toStringTag]
_setToStringTag(Math, 'Math', true);
// 24.3.3 JSON[@@toStringTag]
_setToStringTag(_global.JSON, 'JSON', true);

// call something on iterator step with safe closing on error

var _iterCall = function (iterator, fn, value, entries) {
  try {
    return entries ? fn(_anObject(value)[0], value[1]) : fn(value);
  // 7.4.6 IteratorClose(iterator, completion)
  } catch (e) {
    var ret = iterator['return'];
    if (ret !== undefined) _anObject(ret.call(iterator));
    throw e;
  }
};

var _iterators = {};

// check on default Array iterator

var ITERATOR = _wks('iterator');
var ArrayProto = Array.prototype;

var _isArrayIter = function (it) {
  return it !== undefined && (_iterators.Array === it || ArrayProto[ITERATOR] === it);
};

var _createProperty = function (object, index, value) {
  if (index in object) _objectDp.f(object, index, _propertyDesc(0, value));
  else object[index] = value;
};

// getting tag from 19.1.3.6 Object.prototype.toString()

var TAG$1 = _wks('toStringTag');
// ES3 wrong here
var ARG = _cof(function () { return arguments; }()) == 'Arguments';

// fallback for IE11 Script Access Denied error
var tryGet = function (it, key) {
  try {
    return it[key];
  } catch (e) { /* empty */ }
};

var _classof = function (it) {
  var O, T, B;
  return it === undefined ? 'Undefined' : it === null ? 'Null'
    // @@toStringTag case
    : typeof (T = tryGet(O = Object(it), TAG$1)) == 'string' ? T
    // builtinTag case
    : ARG ? _cof(O)
    // ES3 arguments fallback
    : (B = _cof(O)) == 'Object' && typeof O.callee == 'function' ? 'Arguments' : B;
};

var ITERATOR$1 = _wks('iterator');

var core_getIteratorMethod = _core.getIteratorMethod = function (it) {
  if (it != undefined) return it[ITERATOR$1]
    || it['@@iterator']
    || _iterators[_classof(it)];
};

var ITERATOR$2 = _wks('iterator');
var SAFE_CLOSING = false;

try {
  var riter = [7][ITERATOR$2]();
  riter['return'] = function () { SAFE_CLOSING = true; };
  // eslint-disable-next-line no-throw-literal
  Array.from(riter, function () { throw 2; });
} catch (e) { /* empty */ }

var _iterDetect = function (exec, skipClosing) {
  if (!skipClosing && !SAFE_CLOSING) return false;
  var safe = false;
  try {
    var arr = [7];
    var iter = arr[ITERATOR$2]();
    iter.next = function () { return { done: safe = true }; };
    arr[ITERATOR$2] = function () { return iter; };
    exec(arr);
  } catch (e) { /* empty */ }
  return safe;
};

_export(_export.S + _export.F * !_iterDetect(function (iter) { Array.from(iter); }), 'Array', {
  // 22.1.2.1 Array.from(arrayLike, mapfn = undefined, thisArg = undefined)
  from: function from(arrayLike /* , mapfn = undefined, thisArg = undefined */) {
    var O = _toObject(arrayLike);
    var C = typeof this == 'function' ? this : Array;
    var aLen = arguments.length;
    var mapfn = aLen > 1 ? arguments[1] : undefined;
    var mapping = mapfn !== undefined;
    var index = 0;
    var iterFn = core_getIteratorMethod(O);
    var length, result, step, iterator;
    if (mapping) mapfn = _ctx(mapfn, aLen > 2 ? arguments[2] : undefined, 2);
    // if object isn't iterable or it's array with default iterator - use simple case
    if (iterFn != undefined && !(C == Array && _isArrayIter(iterFn))) {
      for (iterator = iterFn.call(O), result = new C(); !(step = iterator.next()).done; index++) {
        _createProperty(result, index, mapping ? _iterCall(iterator, mapfn, [step.value, index], true) : step.value);
      }
    } else {
      length = _toLength(O.length);
      for (result = new C(length); length > index; index++) {
        _createProperty(result, index, mapping ? mapfn(O[index], index) : O[index]);
      }
    }
    result.length = index;
    return result;
  }
});

// 21.2.5.3 get RegExp.prototype.flags

var _flags = function () {
  var that = _anObject(this);
  var result = '';
  if (that.global) result += 'g';
  if (that.ignoreCase) result += 'i';
  if (that.multiline) result += 'm';
  if (that.unicode) result += 'u';
  if (that.sticky) result += 'y';
  return result;
};

// 21.2.5.3 get RegExp.prototype.flags()
if (_descriptors && /./g.flags != 'g') _objectDp.f(RegExp.prototype, 'flags', {
  configurable: true,
  get: _flags
});

var TO_STRING = 'toString';
var $toString = /./[TO_STRING];

var define = function (fn) {
  _redefine(RegExp.prototype, TO_STRING, fn, true);
};

// 21.2.5.14 RegExp.prototype.toString()
if (_fails(function () { return $toString.call({ source: 'a', flags: 'b' }) != '/a/b'; })) {
  define(function toString() {
    var R = _anObject(this);
    return '/'.concat(R.source, '/',
      'flags' in R ? R.flags : !_descriptors && R instanceof RegExp ? _flags.call(R) : undefined);
  });
// FF44- RegExp#toString has a wrong name
} else if ($toString.name != TO_STRING) {
  define(function toString() {
    return $toString.call(this);
  });
}

function _arrayLikeToArray(arr, len) {
  if (len == null || len > arr.length) len = arr.length;

  for (var i = 0, arr2 = new Array(len); i < len; i++) {
    arr2[i] = arr[i];
  }

  return arr2;
}

function _arrayWithoutHoles(arr) {
  if (Array.isArray(arr)) return _arrayLikeToArray(arr);
}

function _iterableToArray(iter) {
  if (typeof Symbol !== "undefined" && Symbol.iterator in Object(iter)) return Array.from(iter);
}

function _unsupportedIterableToArray(o, minLen) {
  if (!o) return;
  if (typeof o === "string") return _arrayLikeToArray(o, minLen);
  var n = Object.prototype.toString.call(o).slice(8, -1);
  if (n === "Object" && o.constructor) n = o.constructor.name;
  if (n === "Map" || n === "Set") return Array.from(o);
  if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen);
}

function _nonIterableSpread() {
  throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}

function _toConsumableArray(arr) {
  return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread();
}

function _arrayWithHoles(arr) {
  if (Array.isArray(arr)) return arr;
}

function _iterableToArrayLimit(arr, i) {
  if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return;
  var _arr = [];
  var _n = true;
  var _d = false;
  var _e = undefined;

  try {
    for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) {
      _arr.push(_s.value);

      if (i && _arr.length === i) break;
    }
  } catch (err) {
    _d = true;
    _e = err;
  } finally {
    try {
      if (!_n && _i["return"] != null) _i["return"]();
    } finally {
      if (_d) throw _e;
    }
  }

  return _arr;
}

function _nonIterableRest() {
  throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method.");
}

function _slicedToArray(arr, i) {
  return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest();
}

// true  -> String#at
// false -> String#codePointAt
var _stringAt = function (TO_STRING) {
  return function (that, pos) {
    var s = String(_defined(that));
    var i = _toInteger(pos);
    var l = s.length;
    var a, b;
    if (i < 0 || i >= l) return TO_STRING ? '' : undefined;
    a = s.charCodeAt(i);
    return a < 0xd800 || a > 0xdbff || i + 1 === l || (b = s.charCodeAt(i + 1)) < 0xdc00 || b > 0xdfff
      ? TO_STRING ? s.charAt(i) : a
      : TO_STRING ? s.slice(i, i + 2) : (a - 0xd800 << 10) + (b - 0xdc00) + 0x10000;
  };
};

var IteratorPrototype = {};

// 25.1.2.1.1 %IteratorPrototype%[@@iterator]()
_hide(IteratorPrototype, _wks('iterator'), function () { return this; });

var _iterCreate = function (Constructor, NAME, next) {
  Constructor.prototype = _objectCreate(IteratorPrototype, { next: _propertyDesc(1, next) });
  _setToStringTag(Constructor, NAME + ' Iterator');
};

// 19.1.2.9 / 15.2.3.2 Object.getPrototypeOf(O)


var IE_PROTO$2 = _sharedKey('IE_PROTO');
var ObjectProto$1 = Object.prototype;

var _objectGpo = Object.getPrototypeOf || function (O) {
  O = _toObject(O);
  if (_has(O, IE_PROTO$2)) return O[IE_PROTO$2];
  if (typeof O.constructor == 'function' && O instanceof O.constructor) {
    return O.constructor.prototype;
  } return O instanceof Object ? ObjectProto$1 : null;
};

var ITERATOR$3 = _wks('iterator');
var BUGGY = !([].keys && 'next' in [].keys()); // Safari has buggy iterators w/o `next`
var FF_ITERATOR = '@@iterator';
var KEYS = 'keys';
var VALUES = 'values';

var returnThis = function () { return this; };

var _iterDefine = function (Base, NAME, Constructor, next, DEFAULT, IS_SET, FORCED) {
  _iterCreate(Constructor, NAME, next);
  var getMethod = function (kind) {
    if (!BUGGY && kind in proto) return proto[kind];
    switch (kind) {
      case KEYS: return function keys() { return new Constructor(this, kind); };
      case VALUES: return function values() { return new Constructor(this, kind); };
    } return function entries() { return new Constructor(this, kind); };
  };
  var TAG = NAME + ' Iterator';
  var DEF_VALUES = DEFAULT == VALUES;
  var VALUES_BUG = false;
  var proto = Base.prototype;
  var $native = proto[ITERATOR$3] || proto[FF_ITERATOR] || DEFAULT && proto[DEFAULT];
  var $default = $native || getMethod(DEFAULT);
  var $entries = DEFAULT ? !DEF_VALUES ? $default : getMethod('entries') : undefined;
  var $anyNative = NAME == 'Array' ? proto.entries || $native : $native;
  var methods, key, IteratorPrototype;
  // Fix native
  if ($anyNative) {
    IteratorPrototype = _objectGpo($anyNative.call(new Base()));
    if (IteratorPrototype !== Object.prototype && IteratorPrototype.next) {
      // Set @@toStringTag to native iterators
      _setToStringTag(IteratorPrototype, TAG, true);
      // fix for some old engines
      if ( typeof IteratorPrototype[ITERATOR$3] != 'function') _hide(IteratorPrototype, ITERATOR$3, returnThis);
    }
  }
  // fix Array#{values, @@iterator}.name in V8 / FF
  if (DEF_VALUES && $native && $native.name !== VALUES) {
    VALUES_BUG = true;
    $default = function values() { return $native.call(this); };
  }
  // Define iterator
  if ( (BUGGY || VALUES_BUG || !proto[ITERATOR$3])) {
    _hide(proto, ITERATOR$3, $default);
  }
  // Plug for library
  _iterators[NAME] = $default;
  _iterators[TAG] = returnThis;
  if (DEFAULT) {
    methods = {
      values: DEF_VALUES ? $default : getMethod(VALUES),
      keys: IS_SET ? $default : getMethod(KEYS),
      entries: $entries
    };
    if (FORCED) for (key in methods) {
      if (!(key in proto)) _redefine(proto, key, methods[key]);
    } else _export(_export.P + _export.F * (BUGGY || VALUES_BUG), NAME, methods);
  }
  return methods;
};

var $at = _stringAt(true);

// 21.1.3.27 String.prototype[@@iterator]()
_iterDefine(String, 'String', function (iterated) {
  this._t = String(iterated); // target
  this._i = 0;                // next index
// 21.1.5.2.1 %StringIteratorPrototype%.next()
}, function () {
  var O = this._t;
  var index = this._i;
  var point;
  if (index >= O.length) return { value: undefined, done: true };
  point = $at(O, index);
  this._i += point.length;
  return { value: point, done: false };
});

var _redefineAll = function (target, src, safe) {
  for (var key in src) _redefine(target, key, src[key], safe);
  return target;
};

var _anInstance = function (it, Constructor, name, forbiddenField) {
  if (!(it instanceof Constructor) || (forbiddenField !== undefined && forbiddenField in it)) {
    throw TypeError(name + ': incorrect invocation!');
  } return it;
};

var _forOf = createCommonjsModule(function (module) {
var BREAK = {};
var RETURN = {};
var exports = module.exports = function (iterable, entries, fn, that, ITERATOR) {
  var iterFn = ITERATOR ? function () { return iterable; } : core_getIteratorMethod(iterable);
  var f = _ctx(fn, that, entries ? 2 : 1);
  var index = 0;
  var length, step, iterator, result;
  if (typeof iterFn != 'function') throw TypeError(iterable + ' is not iterable!');
  // fast case for arrays with default iterator
  if (_isArrayIter(iterFn)) for (length = _toLength(iterable.length); length > index; index++) {
    result = entries ? f(_anObject(step = iterable[index])[0], step[1]) : f(iterable[index]);
    if (result === BREAK || result === RETURN) return result;
  } else for (iterator = iterFn.call(iterable); !(step = iterator.next()).done;) {
    result = _iterCall(iterator, f, step.value, entries);
    if (result === BREAK || result === RETURN) return result;
  }
};
exports.BREAK = BREAK;
exports.RETURN = RETURN;
});

var _iterStep = function (done, value) {
  return { value: value, done: !!done };
};

var SPECIES = _wks('species');

var _setSpecies = function (KEY) {
  var C = _global[KEY];
  if (_descriptors && C && !C[SPECIES]) _objectDp.f(C, SPECIES, {
    configurable: true,
    get: function () { return this; }
  });
};

var _validateCollection = function (it, TYPE) {
  if (!_isObject(it) || it._t !== TYPE) throw TypeError('Incompatible receiver, ' + TYPE + ' required!');
  return it;
};

var dP$2 = _objectDp.f;









var fastKey = _meta.fastKey;

var SIZE = _descriptors ? '_s' : 'size';

var getEntry = function (that, key) {
  // fast case
  var index = fastKey(key);
  var entry;
  if (index !== 'F') return that._i[index];
  // frozen object case
  for (entry = that._f; entry; entry = entry.n) {
    if (entry.k == key) return entry;
  }
};

var _collectionStrong = {
  getConstructor: function (wrapper, NAME, IS_MAP, ADDER) {
    var C = wrapper(function (that, iterable) {
      _anInstance(that, C, NAME, '_i');
      that._t = NAME;         // collection type
      that._i = _objectCreate(null); // index
      that._f = undefined;    // first entry
      that._l = undefined;    // last entry
      that[SIZE] = 0;         // size
      if (iterable != undefined) _forOf(iterable, IS_MAP, that[ADDER], that);
    });
    _redefineAll(C.prototype, {
      // 23.1.3.1 Map.prototype.clear()
      // 23.2.3.2 Set.prototype.clear()
      clear: function clear() {
        for (var that = _validateCollection(this, NAME), data = that._i, entry = that._f; entry; entry = entry.n) {
          entry.r = true;
          if (entry.p) entry.p = entry.p.n = undefined;
          delete data[entry.i];
        }
        that._f = that._l = undefined;
        that[SIZE] = 0;
      },
      // 23.1.3.3 Map.prototype.delete(key)
      // 23.2.3.4 Set.prototype.delete(value)
      'delete': function (key) {
        var that = _validateCollection(this, NAME);
        var entry = getEntry(that, key);
        if (entry) {
          var next = entry.n;
          var prev = entry.p;
          delete that._i[entry.i];
          entry.r = true;
          if (prev) prev.n = next;
          if (next) next.p = prev;
          if (that._f == entry) that._f = next;
          if (that._l == entry) that._l = prev;
          that[SIZE]--;
        } return !!entry;
      },
      // 23.2.3.6 Set.prototype.forEach(callbackfn, thisArg = undefined)
      // 23.1.3.5 Map.prototype.forEach(callbackfn, thisArg = undefined)
      forEach: function forEach(callbackfn /* , that = undefined */) {
        _validateCollection(this, NAME);
        var f = _ctx(callbackfn, arguments.length > 1 ? arguments[1] : undefined, 3);
        var entry;
        while (entry = entry ? entry.n : this._f) {
          f(entry.v, entry.k, this);
          // revert to the last existing entry
          while (entry && entry.r) entry = entry.p;
        }
      },
      // 23.1.3.7 Map.prototype.has(key)
      // 23.2.3.7 Set.prototype.has(value)
      has: function has(key) {
        return !!getEntry(_validateCollection(this, NAME), key);
      }
    });
    if (_descriptors) dP$2(C.prototype, 'size', {
      get: function () {
        return _validateCollection(this, NAME)[SIZE];
      }
    });
    return C;
  },
  def: function (that, key, value) {
    var entry = getEntry(that, key);
    var prev, index;
    // change existing entry
    if (entry) {
      entry.v = value;
    // create new entry
    } else {
      that._l = entry = {
        i: index = fastKey(key, true), // <- index
        k: key,                        // <- key
        v: value,                      // <- value
        p: prev = that._l,             // <- previous entry
        n: undefined,                  // <- next entry
        r: false                       // <- removed
      };
      if (!that._f) that._f = entry;
      if (prev) prev.n = entry;
      that[SIZE]++;
      // add to index
      if (index !== 'F') that._i[index] = entry;
    } return that;
  },
  getEntry: getEntry,
  setStrong: function (C, NAME, IS_MAP) {
    // add .keys, .values, .entries, [@@iterator]
    // 23.1.3.4, 23.1.3.8, 23.1.3.11, 23.1.3.12, 23.2.3.5, 23.2.3.8, 23.2.3.10, 23.2.3.11
    _iterDefine(C, NAME, function (iterated, kind) {
      this._t = _validateCollection(iterated, NAME); // target
      this._k = kind;                     // kind
      this._l = undefined;                // previous
    }, function () {
      var that = this;
      var kind = that._k;
      var entry = that._l;
      // revert to the last existing entry
      while (entry && entry.r) entry = entry.p;
      // get next entry
      if (!that._t || !(that._l = entry = entry ? entry.n : that._t._f)) {
        // or finish the iteration
        that._t = undefined;
        return _iterStep(1);
      }
      // return step by kind
      if (kind == 'keys') return _iterStep(0, entry.k);
      if (kind == 'values') return _iterStep(0, entry.v);
      return _iterStep(0, [entry.k, entry.v]);
    }, IS_MAP ? 'entries' : 'values', !IS_MAP, true);

    // add [@@species], 23.1.2.2, 23.2.2.2
    _setSpecies(NAME);
  }
};

// Works with __proto__ only. Old v8 can't work with null proto objects.
/* eslint-disable no-proto */


var check = function (O, proto) {
  _anObject(O);
  if (!_isObject(proto) && proto !== null) throw TypeError(proto + ": can't set as prototype!");
};
var _setProto = {
  set: Object.setPrototypeOf || ('__proto__' in {} ? // eslint-disable-line
    function (test, buggy, set) {
      try {
        set = _ctx(Function.call, _objectGopd.f(Object.prototype, '__proto__').set, 2);
        set(test, []);
        buggy = !(test instanceof Array);
      } catch (e) { buggy = true; }
      return function setPrototypeOf(O, proto) {
        check(O, proto);
        if (buggy) O.__proto__ = proto;
        else set(O, proto);
        return O;
      };
    }({}, false) : undefined),
  check: check
};

var setPrototypeOf = _setProto.set;
var _inheritIfRequired = function (that, target, C) {
  var S = target.constructor;
  var P;
  if (S !== C && typeof S == 'function' && (P = S.prototype) !== C.prototype && _isObject(P) && setPrototypeOf) {
    setPrototypeOf(that, P);
  } return that;
};

var _collection = function (NAME, wrapper, methods, common, IS_MAP, IS_WEAK) {
  var Base = _global[NAME];
  var C = Base;
  var ADDER = IS_MAP ? 'set' : 'add';
  var proto = C && C.prototype;
  var O = {};
  var fixMethod = function (KEY) {
    var fn = proto[KEY];
    _redefine(proto, KEY,
      KEY == 'delete' ? function (a) {
        return IS_WEAK && !_isObject(a) ? false : fn.call(this, a === 0 ? 0 : a);
      } : KEY == 'has' ? function has(a) {
        return IS_WEAK && !_isObject(a) ? false : fn.call(this, a === 0 ? 0 : a);
      } : KEY == 'get' ? function get(a) {
        return IS_WEAK && !_isObject(a) ? undefined : fn.call(this, a === 0 ? 0 : a);
      } : KEY == 'add' ? function add(a) { fn.call(this, a === 0 ? 0 : a); return this; }
        : function set(a, b) { fn.call(this, a === 0 ? 0 : a, b); return this; }
    );
  };
  if (typeof C != 'function' || !(IS_WEAK || proto.forEach && !_fails(function () {
    new C().entries().next();
  }))) {
    // create collection constructor
    C = common.getConstructor(wrapper, NAME, IS_MAP, ADDER);
    _redefineAll(C.prototype, methods);
    _meta.NEED = true;
  } else {
    var instance = new C();
    // early implementations not supports chaining
    var HASNT_CHAINING = instance[ADDER](IS_WEAK ? {} : -0, 1) != instance;
    // V8 ~  Chromium 40- weak-collections throws on primitives, but should return false
    var THROWS_ON_PRIMITIVES = _fails(function () { instance.has(1); });
    // most early implementations doesn't supports iterables, most modern - not close it correctly
    var ACCEPT_ITERABLES = _iterDetect(function (iter) { new C(iter); }); // eslint-disable-line no-new
    // for early implementations -0 and +0 not the same
    var BUGGY_ZERO = !IS_WEAK && _fails(function () {
      // V8 ~ Chromium 42- fails only with 5+ elements
      var $instance = new C();
      var index = 5;
      while (index--) $instance[ADDER](index, index);
      return !$instance.has(-0);
    });
    if (!ACCEPT_ITERABLES) {
      C = wrapper(function (target, iterable) {
        _anInstance(target, C, NAME);
        var that = _inheritIfRequired(new Base(), target, C);
        if (iterable != undefined) _forOf(iterable, IS_MAP, that[ADDER], that);
        return that;
      });
      C.prototype = proto;
      proto.constructor = C;
    }
    if (THROWS_ON_PRIMITIVES || BUGGY_ZERO) {
      fixMethod('delete');
      fixMethod('has');
      IS_MAP && fixMethod('get');
    }
    if (BUGGY_ZERO || HASNT_CHAINING) fixMethod(ADDER);
    // weak collections should not contains .clear method
    if (IS_WEAK && proto.clear) delete proto.clear;
  }

  _setToStringTag(C, NAME);

  O[NAME] = C;
  _export(_export.G + _export.W + _export.F * (C != Base), O);

  if (!IS_WEAK) common.setStrong(C, NAME, IS_MAP);

  return C;
};

var SET = 'Set';

// 23.2 Set Objects
var es6_set = _collection(SET, function (get) {
  return function Set() { return get(this, arguments.length > 0 ? arguments[0] : undefined); };
}, {
  // 23.2.3.1 Set.prototype.add(value)
  add: function add(value) {
    return _collectionStrong.def(_validateCollection(this, SET), value = value === 0 ? 0 : value, value);
  }
}, _collectionStrong);

// 22.1.3.31 Array.prototype[@@unscopables]
var UNSCOPABLES = _wks('unscopables');
var ArrayProto$1 = Array.prototype;
if (ArrayProto$1[UNSCOPABLES] == undefined) _hide(ArrayProto$1, UNSCOPABLES, {});
var _addToUnscopables = function (key) {
  ArrayProto$1[UNSCOPABLES][key] = true;
};

// 22.1.3.4 Array.prototype.entries()
// 22.1.3.13 Array.prototype.keys()
// 22.1.3.29 Array.prototype.values()
// 22.1.3.30 Array.prototype[@@iterator]()
var es6_array_iterator = _iterDefine(Array, 'Array', function (iterated, kind) {
  this._t = _toIobject(iterated); // target
  this._i = 0;                   // next index
  this._k = kind;                // kind
// 22.1.5.2.1 %ArrayIteratorPrototype%.next()
}, function () {
  var O = this._t;
  var kind = this._k;
  var index = this._i++;
  if (!O || index >= O.length) {
    this._t = undefined;
    return _iterStep(1);
  }
  if (kind == 'keys') return _iterStep(0, index);
  if (kind == 'values') return _iterStep(0, O[index]);
  return _iterStep(0, [index, O[index]]);
}, 'values');

// argumentsList[@@iterator] is %ArrayProto_values% (9.4.4.6, 9.4.4.7)
_iterators.Arguments = _iterators.Array;

_addToUnscopables('keys');
_addToUnscopables('values');
_addToUnscopables('entries');

var ITERATOR$4 = _wks('iterator');
var TO_STRING_TAG = _wks('toStringTag');
var ArrayValues = _iterators.Array;

var DOMIterables = {
  CSSRuleList: true, // TODO: Not spec compliant, should be false.
  CSSStyleDeclaration: false,
  CSSValueList: false,
  ClientRectList: false,
  DOMRectList: false,
  DOMStringList: false,
  DOMTokenList: true,
  DataTransferItemList: false,
  FileList: false,
  HTMLAllCollection: false,
  HTMLCollection: false,
  HTMLFormElement: false,
  HTMLSelectElement: false,
  MediaList: true, // TODO: Not spec compliant, should be false.
  MimeTypeArray: false,
  NamedNodeMap: false,
  NodeList: true,
  PaintRequestList: false,
  Plugin: false,
  PluginArray: false,
  SVGLengthList: false,
  SVGNumberList: false,
  SVGPathSegList: false,
  SVGPointList: false,
  SVGStringList: false,
  SVGTransformList: false,
  SourceBufferList: false,
  StyleSheetList: true, // TODO: Not spec compliant, should be false.
  TextTrackCueList: false,
  TextTrackList: false,
  TouchList: false
};

for (var collections = _objectKeys(DOMIterables), i = 0; i < collections.length; i++) {
  var NAME = collections[i];
  var explicit = DOMIterables[NAME];
  var Collection = _global[NAME];
  var proto = Collection && Collection.prototype;
  var key;
  if (proto) {
    if (!proto[ITERATOR$4]) _hide(proto, ITERATOR$4, ArrayValues);
    if (!proto[TO_STRING_TAG]) _hide(proto, TO_STRING_TAG, NAME);
    _iterators[NAME] = ArrayValues;
    if (explicit) for (key in es6_array_iterator) if (!proto[key]) _redefine(proto, key, es6_array_iterator[key], true);
  }
}

var at = _stringAt(true);

 // `AdvanceStringIndex` abstract operation
// https://tc39.github.io/ecma262/#sec-advancestringindex
var _advanceStringIndex = function (S, index, unicode) {
  return index + (unicode ? at(S, index).length : 1);
};

var builtinExec = RegExp.prototype.exec;

 // `RegExpExec` abstract operation
// https://tc39.github.io/ecma262/#sec-regexpexec
var _regexpExecAbstract = function (R, S) {
  var exec = R.exec;
  if (typeof exec === 'function') {
    var result = exec.call(R, S);
    if (typeof result !== 'object') {
      throw new TypeError('RegExp exec method returned something other than an Object or null');
    }
    return result;
  }
  if (_classof(R) !== 'RegExp') {
    throw new TypeError('RegExp#exec called on incompatible receiver');
  }
  return builtinExec.call(R, S);
};

var nativeExec = RegExp.prototype.exec;
// This always refers to the native implementation, because the
// String#replace polyfill uses ./fix-regexp-well-known-symbol-logic.js,
// which loads this file before patching the method.
var nativeReplace = String.prototype.replace;

var patchedExec = nativeExec;

var LAST_INDEX = 'lastIndex';

var UPDATES_LAST_INDEX_WRONG = (function () {
  var re1 = /a/,
      re2 = /b*/g;
  nativeExec.call(re1, 'a');
  nativeExec.call(re2, 'a');
  return re1[LAST_INDEX] !== 0 || re2[LAST_INDEX] !== 0;
})();

// nonparticipating capturing group, copied from es5-shim's String#split patch.
var NPCG_INCLUDED = /()??/.exec('')[1] !== undefined;

var PATCH = UPDATES_LAST_INDEX_WRONG || NPCG_INCLUDED;

if (PATCH) {
  patchedExec = function exec(str) {
    var re = this;
    var lastIndex, reCopy, match, i;

    if (NPCG_INCLUDED) {
      reCopy = new RegExp('^' + re.source + '$(?!\\s)', _flags.call(re));
    }
    if (UPDATES_LAST_INDEX_WRONG) lastIndex = re[LAST_INDEX];

    match = nativeExec.call(re, str);

    if (UPDATES_LAST_INDEX_WRONG && match) {
      re[LAST_INDEX] = re.global ? match.index + match[0].length : lastIndex;
    }
    if (NPCG_INCLUDED && match && match.length > 1) {
      // Fix browsers whose `exec` methods don't consistently return `undefined`
      // for NPCG, like IE8. NOTE: This doesn' work for /(.?)?/
      // eslint-disable-next-line no-loop-func
      nativeReplace.call(match[0], reCopy, function () {
        for (i = 1; i < arguments.length - 2; i++) {
          if (arguments[i] === undefined) match[i] = undefined;
        }
      });
    }

    return match;
  };
}

var _regexpExec = patchedExec;

_export({
  target: 'RegExp',
  proto: true,
  forced: _regexpExec !== /./.exec
}, {
  exec: _regexpExec
});

var SPECIES$1 = _wks('species');

var REPLACE_SUPPORTS_NAMED_GROUPS = !_fails(function () {
  // #replace needs built-in support for named groups.
  // #match works fine because it just return the exec results, even if it has
  // a "grops" property.
  var re = /./;
  re.exec = function () {
    var result = [];
    result.groups = { a: '7' };
    return result;
  };
  return ''.replace(re, '$<a>') !== '7';
});

var SPLIT_WORKS_WITH_OVERWRITTEN_EXEC = (function () {
  // Chrome 51 has a buggy "split" implementation when RegExp#exec !== nativeExec
  var re = /(?:)/;
  var originalExec = re.exec;
  re.exec = function () { return originalExec.apply(this, arguments); };
  var result = 'ab'.split(re);
  return result.length === 2 && result[0] === 'a' && result[1] === 'b';
})();

var _fixReWks = function (KEY, length, exec) {
  var SYMBOL = _wks(KEY);

  var DELEGATES_TO_SYMBOL = !_fails(function () {
    // String methods call symbol-named RegEp methods
    var O = {};
    O[SYMBOL] = function () { return 7; };
    return ''[KEY](O) != 7;
  });

  var DELEGATES_TO_EXEC = DELEGATES_TO_SYMBOL ? !_fails(function () {
    // Symbol-named RegExp methods call .exec
    var execCalled = false;
    var re = /a/;
    re.exec = function () { execCalled = true; return null; };
    if (KEY === 'split') {
      // RegExp[@@split] doesn't call the regex's exec method, but first creates
      // a new one. We need to return the patched regex when creating the new one.
      re.constructor = {};
      re.constructor[SPECIES$1] = function () { return re; };
    }
    re[SYMBOL]('');
    return !execCalled;
  }) : undefined;

  if (
    !DELEGATES_TO_SYMBOL ||
    !DELEGATES_TO_EXEC ||
    (KEY === 'replace' && !REPLACE_SUPPORTS_NAMED_GROUPS) ||
    (KEY === 'split' && !SPLIT_WORKS_WITH_OVERWRITTEN_EXEC)
  ) {
    var nativeRegExpMethod = /./[SYMBOL];
    var fns = exec(
      _defined,
      SYMBOL,
      ''[KEY],
      function maybeCallNative(nativeMethod, regexp, str, arg2, forceStringMethod) {
        if (regexp.exec === _regexpExec) {
          if (DELEGATES_TO_SYMBOL && !forceStringMethod) {
            // The native String method already delegates to @@method (this
            // polyfilled function), leasing to infinite recursion.
            // We avoid it by directly calling the native @@method method.
            return { done: true, value: nativeRegExpMethod.call(regexp, str, arg2) };
          }
          return { done: true, value: nativeMethod.call(str, regexp, arg2) };
        }
        return { done: false };
      }
    );
    var strfn = fns[0];
    var rxfn = fns[1];

    _redefine(String.prototype, KEY, strfn);
    _hide(RegExp.prototype, SYMBOL, length == 2
      // 21.2.5.8 RegExp.prototype[@@replace](string, replaceValue)
      // 21.2.5.11 RegExp.prototype[@@split](string, limit)
      ? function (string, arg) { return rxfn.call(string, this, arg); }
      // 21.2.5.6 RegExp.prototype[@@match](string)
      // 21.2.5.9 RegExp.prototype[@@search](string)
      : function (string) { return rxfn.call(string, this); }
    );
  }
};

var max$1 = Math.max;
var min$2 = Math.min;
var floor$1 = Math.floor;
var SUBSTITUTION_SYMBOLS = /\$([$&`']|\d\d?|<[^>]*>)/g;
var SUBSTITUTION_SYMBOLS_NO_NAMED = /\$([$&`']|\d\d?)/g;

var maybeToString = function (it) {
  return it === undefined ? it : String(it);
};

// @@replace logic
_fixReWks('replace', 2, function (defined, REPLACE, $replace, maybeCallNative) {
  return [
    // `String.prototype.replace` method
    // https://tc39.github.io/ecma262/#sec-string.prototype.replace
    function replace(searchValue, replaceValue) {
      var O = defined(this);
      var fn = searchValue == undefined ? undefined : searchValue[REPLACE];
      return fn !== undefined
        ? fn.call(searchValue, O, replaceValue)
        : $replace.call(String(O), searchValue, replaceValue);
    },
    // `RegExp.prototype[@@replace]` method
    // https://tc39.github.io/ecma262/#sec-regexp.prototype-@@replace
    function (regexp, replaceValue) {
      var res = maybeCallNative($replace, regexp, this, replaceValue);
      if (res.done) return res.value;

      var rx = _anObject(regexp);
      var S = String(this);
      var functionalReplace = typeof replaceValue === 'function';
      if (!functionalReplace) replaceValue = String(replaceValue);
      var global = rx.global;
      if (global) {
        var fullUnicode = rx.unicode;
        rx.lastIndex = 0;
      }
      var results = [];
      while (true) {
        var result = _regexpExecAbstract(rx, S);
        if (result === null) break;
        results.push(result);
        if (!global) break;
        var matchStr = String(result[0]);
        if (matchStr === '') rx.lastIndex = _advanceStringIndex(S, _toLength(rx.lastIndex), fullUnicode);
      }
      var accumulatedResult = '';
      var nextSourcePosition = 0;
      for (var i = 0; i < results.length; i++) {
        result = results[i];
        var matched = String(result[0]);
        var position = max$1(min$2(_toInteger(result.index), S.length), 0);
        var captures = [];
        // NOTE: This is equivalent to
        //   captures = result.slice(1).map(maybeToString)
        // but for some reason `nativeSlice.call(result, 1, result.length)` (called in
        // the slice polyfill when slicing native arrays) "doesn't work" in safari 9 and
        // causes a crash (https://pastebin.com/N21QzeQA) when trying to debug it.
        for (var j = 1; j < result.length; j++) captures.push(maybeToString(result[j]));
        var namedCaptures = result.groups;
        if (functionalReplace) {
          var replacerArgs = [matched].concat(captures, position, S);
          if (namedCaptures !== undefined) replacerArgs.push(namedCaptures);
          var replacement = String(replaceValue.apply(undefined, replacerArgs));
        } else {
          replacement = getSubstitution(matched, S, position, captures, namedCaptures, replaceValue);
        }
        if (position >= nextSourcePosition) {
          accumulatedResult += S.slice(nextSourcePosition, position) + replacement;
          nextSourcePosition = position + matched.length;
        }
      }
      return accumulatedResult + S.slice(nextSourcePosition);
    }
  ];

    // https://tc39.github.io/ecma262/#sec-getsubstitution
  function getSubstitution(matched, str, position, captures, namedCaptures, replacement) {
    var tailPos = position + matched.length;
    var m = captures.length;
    var symbols = SUBSTITUTION_SYMBOLS_NO_NAMED;
    if (namedCaptures !== undefined) {
      namedCaptures = _toObject(namedCaptures);
      symbols = SUBSTITUTION_SYMBOLS;
    }
    return $replace.call(replacement, symbols, function (match, ch) {
      var capture;
      switch (ch.charAt(0)) {
        case '$': return '$';
        case '&': return matched;
        case '`': return str.slice(0, position);
        case "'": return str.slice(tailPos);
        case '<':
          capture = namedCaptures[ch.slice(1, -1)];
          break;
        default: // \d\d?
          var n = +ch;
          if (n === 0) return match;
          if (n > m) {
            var f = floor$1(n / 10);
            if (f === 0) return match;
            if (f <= m) return captures[f - 1] === undefined ? ch.charAt(1) : captures[f - 1] + ch.charAt(1);
            return match;
          }
          capture = captures[n - 1];
      }
      return capture === undefined ? '' : capture;
    });
  }
});

// @@match logic
_fixReWks('match', 1, function (defined, MATCH, $match, maybeCallNative) {
  return [
    // `String.prototype.match` method
    // https://tc39.github.io/ecma262/#sec-string.prototype.match
    function match(regexp) {
      var O = defined(this);
      var fn = regexp == undefined ? undefined : regexp[MATCH];
      return fn !== undefined ? fn.call(regexp, O) : new RegExp(regexp)[MATCH](String(O));
    },
    // `RegExp.prototype[@@match]` method
    // https://tc39.github.io/ecma262/#sec-regexp.prototype-@@match
    function (regexp) {
      var res = maybeCallNative($match, regexp, this);
      if (res.done) return res.value;
      var rx = _anObject(regexp);
      var S = String(this);
      if (!rx.global) return _regexpExecAbstract(rx, S);
      var fullUnicode = rx.unicode;
      rx.lastIndex = 0;
      var A = [];
      var n = 0;
      var result;
      while ((result = _regexpExecAbstract(rx, S)) !== null) {
        var matchStr = String(result[0]);
        A[n] = matchStr;
        if (matchStr === '') rx.lastIndex = _advanceStringIndex(S, _toLength(rx.lastIndex), fullUnicode);
        n++;
      }
      return n === 0 ? null : A;
    }
  ];
});

var uigradients = [
	{
		name: "Blu",
		colors: [
			"#00416A",
			"#E4E5E6"
		]
	},
	{
		name: "Ver",
		colors: [
			"#FFE000",
			"#799F0C"
		]
	},
	{
		name: "Ver Black",
		colors: [
			"#F7F8F8",
			"#ACBB78"
		]
	},
	{
		name: "Combi",
		colors: [
			"#00416A",
			"#799F0C",
			"#FFE000"
		]
	},
	{
		name: "Anwar",
		colors: [
			"#334d50",
			"#cbcaa5"
		]
	},
	{
		name: "Bluelagoo",
		colors: [
			"#0052D4",
			"#4364F7",
			"#6FB1FC"
		]
	},
	{
		name: "Lunada",
		colors: [
			"#5433FF",
			"#20BDFF",
			"#A5FECB"
		]
	},
	{
		name: "Reaqua",
		colors: [
			"#799F0C",
			"#ACBB78"
		]
	},
	{
		name: "Mango",
		colors: [
			"#ffe259",
			"#ffa751"
		]
	},
	{
		name: "Bupe",
		colors: [
			"#00416A",
			"#E4E5E6"
		]
	},
	{
		name: "Rea",
		colors: [
			"#FFE000",
			"#799F0C"
		]
	},
	{
		name: "Windy",
		colors: [
			"#acb6e5",
			"#86fde8"
		]
	},
	{
		name: "Royal Blue",
		colors: [
			"#536976",
			"#292E49"
		]
	},
	{
		name: "Royal Blue + Petrol",
		colors: [
			"#BBD2C5",
			"#536976",
			"#292E49"
		]
	},
	{
		name: "Copper",
		colors: [
			"#B79891",
			"#94716B"
		]
	},
	{
		name: "Anamnisar",
		colors: [
			"#9796f0",
			"#fbc7d4"
		]
	},
	{
		name: "Petrol",
		colors: [
			"#BBD2C5",
			"#536976"
		]
	},
	{
		name: "Sky",
		colors: [
			"#076585",
			"#fff"
		]
	},
	{
		name: "Sel",
		colors: [
			"#00467F",
			"#A5CC82"
		]
	},
	{
		name: "Skyline",
		colors: [
			"#1488CC",
			"#2B32B2"
		]
	},
	{
		name: "DIMIGO",
		colors: [
			"#ec008c",
			"#fc6767"
		]
	},
	{
		name: "Purple Love",
		colors: [
			"#cc2b5e",
			"#753a88"
		]
	},
	{
		name: "Sexy Blue",
		colors: [
			"#2193b0",
			"#6dd5ed"
		]
	},
	{
		name: "Blooker20",
		colors: [
			"#e65c00",
			"#F9D423"
		]
	},
	{
		name: "Sea Blue",
		colors: [
			"#2b5876",
			"#4e4376"
		]
	},
	{
		name: "Nimvelo",
		colors: [
			"#314755",
			"#26a0da"
		]
	},
	{
		name: "Hazel",
		colors: [
			"#77A1D3",
			"#79CBCA",
			"#E684AE"
		]
	},
	{
		name: "Noon to Dusk",
		colors: [
			"#ff6e7f",
			"#bfe9ff"
		]
	},
	{
		name: "YouTube",
		colors: [
			"#e52d27",
			"#b31217"
		]
	},
	{
		name: "Cool Brown",
		colors: [
			"#603813",
			"#b29f94"
		]
	},
	{
		name: "Harmonic Energy",
		colors: [
			"#16A085",
			"#F4D03F"
		]
	},
	{
		name: "Playing with Reds",
		colors: [
			"#D31027",
			"#EA384D"
		]
	},
	{
		name: "Sunny Days",
		colors: [
			"#EDE574",
			"#E1F5C4"
		]
	},
	{
		name: "Green Beach",
		colors: [
			"#02AAB0",
			"#00CDAC"
		]
	},
	{
		name: "Intuitive Purple",
		colors: [
			"#DA22FF",
			"#9733EE"
		]
	},
	{
		name: "Emerald Water",
		colors: [
			"#348F50",
			"#56B4D3"
		]
	},
	{
		name: "Lemon Twist",
		colors: [
			"#3CA55C",
			"#B5AC49"
		]
	},
	{
		name: "Monte Carlo",
		colors: [
			"#CC95C0",
			"#DBD4B4",
			"#7AA1D2"
		]
	},
	{
		name: "Horizon",
		colors: [
			"#003973",
			"#E5E5BE"
		]
	},
	{
		name: "Rose Water",
		colors: [
			"#E55D87",
			"#5FC3E4"
		]
	},
	{
		name: "Frozen",
		colors: [
			"#403B4A",
			"#E7E9BB"
		]
	},
	{
		name: "Mango Pulp",
		colors: [
			"#F09819",
			"#EDDE5D"
		]
	},
	{
		name: "Bloody Mary",
		colors: [
			"#FF512F",
			"#DD2476"
		]
	},
	{
		name: "Aubergine",
		colors: [
			"#AA076B",
			"#61045F"
		]
	},
	{
		name: "Aqua Marine",
		colors: [
			"#1A2980",
			"#26D0CE"
		]
	},
	{
		name: "Sunrise",
		colors: [
			"#FF512F",
			"#F09819"
		]
	},
	{
		name: "Purple Paradise",
		colors: [
			"#1D2B64",
			"#F8CDDA"
		]
	},
	{
		name: "Stripe",
		colors: [
			"#1FA2FF",
			"#12D8FA",
			"#A6FFCB"
		]
	},
	{
		name: "Sea Weed",
		colors: [
			"#4CB8C4",
			"#3CD3AD"
		]
	},
	{
		name: "Pinky",
		colors: [
			"#DD5E89",
			"#F7BB97"
		]
	},
	{
		name: "Cherry",
		colors: [
			"#EB3349",
			"#F45C43"
		]
	},
	{
		name: "Mojito",
		colors: [
			"#1D976C",
			"#93F9B9"
		]
	},
	{
		name: "Juicy Orange",
		colors: [
			"#FF8008",
			"#FFC837"
		]
	},
	{
		name: "Mirage",
		colors: [
			"#16222A",
			"#3A6073"
		]
	},
	{
		name: "Steel Gray",
		colors: [
			"#1F1C2C",
			"#928DAB"
		]
	},
	{
		name: "Kashmir",
		colors: [
			"#614385",
			"#516395"
		]
	},
	{
		name: "Electric Violet",
		colors: [
			"#4776E6",
			"#8E54E9"
		]
	},
	{
		name: "Venice Blue",
		colors: [
			"#085078",
			"#85D8CE"
		]
	},
	{
		name: "Bora Bora",
		colors: [
			"#2BC0E4",
			"#EAECC6"
		]
	},
	{
		name: "Moss",
		colors: [
			"#134E5E",
			"#71B280"
		]
	},
	{
		name: "Shroom Haze",
		colors: [
			"#5C258D",
			"#4389A2"
		]
	},
	{
		name: "Mystic",
		colors: [
			"#757F9A",
			"#D7DDE8"
		]
	},
	{
		name: "Midnight City",
		colors: [
			"#232526",
			"#414345"
		]
	},
	{
		name: "Sea Blizz",
		colors: [
			"#1CD8D2",
			"#93EDC7"
		]
	},
	{
		name: "Opa",
		colors: [
			"#3D7EAA",
			"#FFE47A"
		]
	},
	{
		name: "Titanium",
		colors: [
			"#283048",
			"#859398"
		]
	},
	{
		name: "Mantle",
		colors: [
			"#24C6DC",
			"#514A9D"
		]
	},
	{
		name: "Dracula",
		colors: [
			"#DC2424",
			"#4A569D"
		]
	},
	{
		name: "Peach",
		colors: [
			"#ED4264",
			"#FFEDBC"
		]
	},
	{
		name: "Moonrise",
		colors: [
			"#DAE2F8",
			"#D6A4A4"
		]
	},
	{
		name: "Clouds",
		colors: [
			"#ECE9E6",
			"#FFFFFF"
		]
	},
	{
		name: "Stellar",
		colors: [
			"#7474BF",
			"#348AC7"
		]
	},
	{
		name: "Bourbon",
		colors: [
			"#EC6F66",
			"#F3A183"
		]
	},
	{
		name: "Calm Darya",
		colors: [
			"#5f2c82",
			"#49a09d"
		]
	},
	{
		name: "Influenza",
		colors: [
			"#C04848",
			"#480048"
		]
	},
	{
		name: "Shrimpy",
		colors: [
			"#e43a15",
			"#e65245"
		]
	},
	{
		name: "Army",
		colors: [
			"#414d0b",
			"#727a17"
		]
	},
	{
		name: "Miaka",
		colors: [
			"#FC354C",
			"#0ABFBC"
		]
	},
	{
		name: "Pinot Noir",
		colors: [
			"#4b6cb7",
			"#182848"
		]
	},
	{
		name: "Day Tripper",
		colors: [
			"#f857a6",
			"#ff5858"
		]
	},
	{
		name: "Namn",
		colors: [
			"#a73737",
			"#7a2828"
		]
	},
	{
		name: "Blurry Beach",
		colors: [
			"#d53369",
			"#cbad6d"
		]
	},
	{
		name: "Vasily",
		colors: [
			"#e9d362",
			"#333333"
		]
	},
	{
		name: "A Lost Memory",
		colors: [
			"#DE6262",
			"#FFB88C"
		]
	},
	{
		name: "Petrichor",
		colors: [
			"#666600",
			"#999966"
		]
	},
	{
		name: "Jonquil",
		colors: [
			"#FFEEEE",
			"#DDEFBB"
		]
	},
	{
		name: "Sirius Tamed",
		colors: [
			"#EFEFBB",
			"#D4D3DD"
		]
	},
	{
		name: "Kyoto",
		colors: [
			"#c21500",
			"#ffc500"
		]
	},
	{
		name: "Misty Meadow",
		colors: [
			"#215f00",
			"#e4e4d9"
		]
	},
	{
		name: "Aqualicious",
		colors: [
			"#50C9C3",
			"#96DEDA"
		]
	},
	{
		name: "Moor",
		colors: [
			"#616161",
			"#9bc5c3"
		]
	},
	{
		name: "Almost",
		colors: [
			"#ddd6f3",
			"#faaca8"
		]
	},
	{
		name: "Forever Lost",
		colors: [
			"#5D4157",
			"#A8CABA"
		]
	},
	{
		name: "Winter",
		colors: [
			"#E6DADA",
			"#274046"
		]
	},
	{
		name: "Nelson",
		colors: [
			"#f2709c",
			"#ff9472"
		]
	},
	{
		name: "Autumn",
		colors: [
			"#DAD299",
			"#B0DAB9"
		]
	},
	{
		name: "Candy",
		colors: [
			"#D3959B",
			"#BFE6BA"
		]
	},
	{
		name: "Reef",
		colors: [
			"#00d2ff",
			"#3a7bd5"
		]
	},
	{
		name: "The Strain",
		colors: [
			"#870000",
			"#190A05"
		]
	},
	{
		name: "Dirty Fog",
		colors: [
			"#B993D6",
			"#8CA6DB"
		]
	},
	{
		name: "Earthly",
		colors: [
			"#649173",
			"#DBD5A4"
		]
	},
	{
		name: "Virgin",
		colors: [
			"#C9FFBF",
			"#FFAFBD"
		]
	},
	{
		name: "Ash",
		colors: [
			"#606c88",
			"#3f4c6b"
		]
	},
	{
		name: "Cherryblossoms",
		colors: [
			"#FBD3E9",
			"#BB377D"
		]
	},
	{
		name: "Parklife",
		colors: [
			"#ADD100",
			"#7B920A"
		]
	},
	{
		name: "Dance To Forget",
		colors: [
			"#FF4E50",
			"#F9D423"
		]
	},
	{
		name: "Starfall",
		colors: [
			"#F0C27B",
			"#4B1248"
		]
	},
	{
		name: "Red Mist",
		colors: [
			"#000000",
			"#e74c3c"
		]
	},
	{
		name: "Teal Love",
		colors: [
			"#AAFFA9",
			"#11FFBD"
		]
	},
	{
		name: "Neon Life",
		colors: [
			"#B3FFAB",
			"#12FFF7"
		]
	},
	{
		name: "Man of Steel",
		colors: [
			"#780206",
			"#061161"
		]
	},
	{
		name: "Amethyst",
		colors: [
			"#9D50BB",
			"#6E48AA"
		]
	},
	{
		name: "Cheer Up Emo Kid",
		colors: [
			"#556270",
			"#FF6B6B"
		]
	},
	{
		name: "Shore",
		colors: [
			"#70e1f5",
			"#ffd194"
		]
	},
	{
		name: "Facebook Messenger",
		colors: [
			"#00c6ff",
			"#0072ff"
		]
	},
	{
		name: "SoundCloud",
		colors: [
			"#fe8c00",
			"#f83600"
		]
	},
	{
		name: "Behongo",
		colors: [
			"#52c234",
			"#061700"
		]
	},
	{
		name: "ServQuick",
		colors: [
			"#485563",
			"#29323c"
		]
	},
	{
		name: "Friday",
		colors: [
			"#83a4d4",
			"#b6fbff"
		]
	},
	{
		name: "Martini",
		colors: [
			"#FDFC47",
			"#24FE41"
		]
	},
	{
		name: "Metallic Toad",
		colors: [
			"#abbaab",
			"#ffffff"
		]
	},
	{
		name: "Between The Clouds",
		colors: [
			"#73C8A9",
			"#373B44"
		]
	},
	{
		name: "Crazy Orange I",
		colors: [
			"#D38312",
			"#A83279"
		]
	},
	{
		name: "Hersheys",
		colors: [
			"#1e130c",
			"#9a8478"
		]
	},
	{
		name: "Talking To Mice Elf",
		colors: [
			"#948E99",
			"#2E1437"
		]
	},
	{
		name: "Purple Bliss",
		colors: [
			"#360033",
			"#0b8793"
		]
	},
	{
		name: "Predawn",
		colors: [
			"#FFA17F",
			"#00223E"
		]
	},
	{
		name: "Endless River",
		colors: [
			"#43cea2",
			"#185a9d"
		]
	},
	{
		name: "Pastel Orange at the Sun",
		colors: [
			"#ffb347",
			"#ffcc33"
		]
	},
	{
		name: "Twitch",
		colors: [
			"#6441A5",
			"#2a0845"
		]
	},
	{
		name: "Atlas",
		colors: [
			"#FEAC5E",
			"#C779D0",
			"#4BC0C8"
		]
	},
	{
		name: "Instagram",
		colors: [
			"#833ab4",
			"#fd1d1d",
			"#fcb045"
		]
	},
	{
		name: "Flickr",
		colors: [
			"#ff0084",
			"#33001b"
		]
	},
	{
		name: "Vine",
		colors: [
			"#00bf8f",
			"#001510"
		]
	},
	{
		name: "Turquoise flow",
		colors: [
			"#136a8a",
			"#267871"
		]
	},
	{
		name: "Portrait",
		colors: [
			"#8e9eab",
			"#eef2f3"
		]
	},
	{
		name: "Virgin America",
		colors: [
			"#7b4397",
			"#dc2430"
		]
	},
	{
		name: "Koko Caramel",
		colors: [
			"#D1913C",
			"#FFD194"
		]
	},
	{
		name: "Fresh Turboscent",
		colors: [
			"#F1F2B5",
			"#135058"
		]
	},
	{
		name: "Green to dark",
		colors: [
			"#6A9113",
			"#141517"
		]
	},
	{
		name: "Ukraine",
		colors: [
			"#004FF9",
			"#FFF94C"
		]
	},
	{
		name: "Curiosity blue",
		colors: [
			"#525252",
			"#3d72b4"
		]
	},
	{
		name: "Dark Knight",
		colors: [
			"#BA8B02",
			"#181818"
		]
	},
	{
		name: "Piglet",
		colors: [
			"#ee9ca7",
			"#ffdde1"
		]
	},
	{
		name: "Lizard",
		colors: [
			"#304352",
			"#d7d2cc"
		]
	},
	{
		name: "Sage Persuasion",
		colors: [
			"#CCCCB2",
			"#757519"
		]
	},
	{
		name: "Between Night and Day",
		colors: [
			"#2c3e50",
			"#3498db"
		]
	},
	{
		name: "Timber",
		colors: [
			"#fc00ff",
			"#00dbde"
		]
	},
	{
		name: "Passion",
		colors: [
			"#e53935",
			"#e35d5b"
		]
	},
	{
		name: "Clear Sky",
		colors: [
			"#005C97",
			"#363795"
		]
	},
	{
		name: "Master Card",
		colors: [
			"#f46b45",
			"#eea849"
		]
	},
	{
		name: "Back To Earth",
		colors: [
			"#00C9FF",
			"#92FE9D"
		]
	},
	{
		name: "Deep Purple",
		colors: [
			"#673AB7",
			"#512DA8"
		]
	},
	{
		name: "Little Leaf",
		colors: [
			"#76b852",
			"#8DC26F"
		]
	},
	{
		name: "Netflix",
		colors: [
			"#8E0E00",
			"#1F1C18"
		]
	},
	{
		name: "Light Orange",
		colors: [
			"#FFB75E",
			"#ED8F03"
		]
	},
	{
		name: "Green and Blue",
		colors: [
			"#c2e59c",
			"#64b3f4"
		]
	},
	{
		name: "Poncho",
		colors: [
			"#403A3E",
			"#BE5869"
		]
	},
	{
		name: "Back to the Future",
		colors: [
			"#C02425",
			"#F0CB35"
		]
	},
	{
		name: "Blush",
		colors: [
			"#B24592",
			"#F15F79"
		]
	},
	{
		name: "Inbox",
		colors: [
			"#457fca",
			"#5691c8"
		]
	},
	{
		name: "Purplin",
		colors: [
			"#6a3093",
			"#a044ff"
		]
	},
	{
		name: "Pale Wood",
		colors: [
			"#eacda3",
			"#d6ae7b"
		]
	},
	{
		name: "Haikus",
		colors: [
			"#fd746c",
			"#ff9068"
		]
	},
	{
		name: "Pizelex",
		colors: [
			"#114357",
			"#F29492"
		]
	},
	{
		name: "Joomla",
		colors: [
			"#1e3c72",
			"#2a5298"
		]
	},
	{
		name: "Christmas",
		colors: [
			"#2F7336",
			"#AA3A38"
		]
	},
	{
		name: "Minnesota Vikings",
		colors: [
			"#5614B0",
			"#DBD65C"
		]
	},
	{
		name: "Miami Dolphins",
		colors: [
			"#4DA0B0",
			"#D39D38"
		]
	},
	{
		name: "Forest",
		colors: [
			"#5A3F37",
			"#2C7744"
		]
	},
	{
		name: "Nighthawk",
		colors: [
			"#2980b9",
			"#2c3e50"
		]
	},
	{
		name: "Superman",
		colors: [
			"#0099F7",
			"#F11712"
		]
	},
	{
		name: "Suzy",
		colors: [
			"#834d9b",
			"#d04ed6"
		]
	},
	{
		name: "Dark Skies",
		colors: [
			"#4B79A1",
			"#283E51"
		]
	},
	{
		name: "Deep Space",
		colors: [
			"#000000",
			"#434343"
		]
	},
	{
		name: "Decent",
		colors: [
			"#4CA1AF",
			"#C4E0E5"
		]
	},
	{
		name: "Colors Of Sky",
		colors: [
			"#E0EAFC",
			"#CFDEF3"
		]
	},
	{
		name: "Purple White",
		colors: [
			"#BA5370",
			"#F4E2D8"
		]
	},
	{
		name: "Ali",
		colors: [
			"#ff4b1f",
			"#1fddff"
		]
	},
	{
		name: "Alihossein",
		colors: [
			"#f7ff00",
			"#db36a4"
		]
	},
	{
		name: "Shahabi",
		colors: [
			"#a80077",
			"#66ff00"
		]
	},
	{
		name: "Red Ocean",
		colors: [
			"#1D4350",
			"#A43931"
		]
	},
	{
		name: "Tranquil",
		colors: [
			"#EECDA3",
			"#EF629F"
		]
	},
	{
		name: "Transfile",
		colors: [
			"#16BFFD",
			"#CB3066"
		]
	},
	{
		name: "Sylvia",
		colors: [
			"#ff4b1f",
			"#ff9068"
		]
	},
	{
		name: "Sweet Morning",
		colors: [
			"#FF5F6D",
			"#FFC371"
		]
	},
	{
		name: "Politics",
		colors: [
			"#2196f3",
			"#f44336"
		]
	},
	{
		name: "Bright Vault",
		colors: [
			"#00d2ff",
			"#928DAB"
		]
	},
	{
		name: "Solid Vault",
		colors: [
			"#3a7bd5",
			"#3a6073"
		]
	},
	{
		name: "Sunset",
		colors: [
			"#0B486B",
			"#F56217"
		]
	},
	{
		name: "Grapefruit Sunset",
		colors: [
			"#e96443",
			"#904e95"
		]
	},
	{
		name: "Deep Sea Space",
		colors: [
			"#2C3E50",
			"#4CA1AF"
		]
	},
	{
		name: "Dusk",
		colors: [
			"#2C3E50",
			"#FD746C"
		]
	},
	{
		name: "Minimal Red",
		colors: [
			"#F00000",
			"#DC281E"
		]
	},
	{
		name: "Royal",
		colors: [
			"#141E30",
			"#243B55"
		]
	},
	{
		name: "Mauve",
		colors: [
			"#42275a",
			"#734b6d"
		]
	},
	{
		name: "Frost",
		colors: [
			"#000428",
			"#004e92"
		]
	},
	{
		name: "Lush",
		colors: [
			"#56ab2f",
			"#a8e063"
		]
	},
	{
		name: "Firewatch",
		colors: [
			"#cb2d3e",
			"#ef473a"
		]
	},
	{
		name: "Sherbert",
		colors: [
			"#f79d00",
			"#64f38c"
		]
	},
	{
		name: "Blood Red",
		colors: [
			"#f85032",
			"#e73827"
		]
	},
	{
		name: "Sun on the Horizon",
		colors: [
			"#fceabb",
			"#f8b500"
		]
	},
	{
		name: "IIIT Delhi",
		colors: [
			"#808080",
			"#3fada8"
		]
	},
	{
		name: "Jupiter",
		colors: [
			"#ffd89b",
			"#19547b"
		]
	},
	{
		name: "50 Shades of Grey",
		colors: [
			"#bdc3c7",
			"#2c3e50"
		]
	},
	{
		name: "Dania",
		colors: [
			"#BE93C5",
			"#7BC6CC"
		]
	},
	{
		name: "Limeade",
		colors: [
			"#A1FFCE",
			"#FAFFD1"
		]
	},
	{
		name: "Disco",
		colors: [
			"#4ECDC4",
			"#556270"
		]
	},
	{
		name: "Love Couple",
		colors: [
			"#3a6186",
			"#89253e"
		]
	},
	{
		name: "Azure Pop",
		colors: [
			"#ef32d9",
			"#89fffd"
		]
	},
	{
		name: "Nepal",
		colors: [
			"#de6161",
			"#2657eb"
		]
	},
	{
		name: "Cosmic Fusion",
		colors: [
			"#ff00cc",
			"#333399"
		]
	},
	{
		name: "Snapchat",
		colors: [
			"#fffc00",
			"#ffffff"
		]
	},
	{
		name: "Ed's Sunset Gradient",
		colors: [
			"#ff7e5f",
			"#feb47b"
		]
	},
	{
		name: "Brady Brady Fun Fun",
		colors: [
			"#00c3ff",
			"#ffff1c"
		]
	},
	{
		name: "Black Rosé",
		colors: [
			"#f4c4f3",
			"#fc67fa"
		]
	},
	{
		name: "80's Purple",
		colors: [
			"#41295a",
			"#2F0743"
		]
	},
	{
		name: "Radar",
		colors: [
			"#A770EF",
			"#CF8BF3",
			"#FDB99B"
		]
	},
	{
		name: "Ibiza Sunset",
		colors: [
			"#ee0979",
			"#ff6a00"
		]
	},
	{
		name: "Dawn",
		colors: [
			"#F3904F",
			"#3B4371"
		]
	},
	{
		name: "Mild",
		colors: [
			"#67B26F",
			"#4ca2cd"
		]
	},
	{
		name: "Vice City",
		colors: [
			"#3494E6",
			"#EC6EAD"
		]
	},
	{
		name: "Jaipur",
		colors: [
			"#DBE6F6",
			"#C5796D"
		]
	},
	{
		name: "Jodhpur",
		colors: [
			"#9CECFB",
			"#65C7F7",
			"#0052D4"
		]
	},
	{
		name: "Cocoaa Ice",
		colors: [
			"#c0c0aa",
			"#1cefff"
		]
	},
	{
		name: "EasyMed",
		colors: [
			"#DCE35B",
			"#45B649"
		]
	},
	{
		name: "Rose Colored Lenses",
		colors: [
			"#E8CBC0",
			"#636FA4"
		]
	},
	{
		name: "What lies Beyond",
		colors: [
			"#F0F2F0",
			"#000C40"
		]
	},
	{
		name: "Roseanna",
		colors: [
			"#FFAFBD",
			"#ffc3a0"
		]
	},
	{
		name: "Honey Dew",
		colors: [
			"#43C6AC",
			"#F8FFAE"
		]
	},
	{
		name: "Under the Lake",
		colors: [
			"#093028",
			"#237A57"
		]
	},
	{
		name: "The Blue Lagoon",
		colors: [
			"#43C6AC",
			"#191654"
		]
	},
	{
		name: "Can You Feel The Love Tonight",
		colors: [
			"#4568DC",
			"#B06AB3"
		]
	},
	{
		name: "Very Blue",
		colors: [
			"#0575E6",
			"#021B79"
		]
	},
	{
		name: "Love and Liberty",
		colors: [
			"#200122",
			"#6f0000"
		]
	},
	{
		name: "Orca",
		colors: [
			"#44A08D",
			"#093637"
		]
	},
	{
		name: "Venice",
		colors: [
			"#6190E8",
			"#A7BFE8"
		]
	},
	{
		name: "Pacific Dream",
		colors: [
			"#34e89e",
			"#0f3443"
		]
	},
	{
		name: "Learning and Leading",
		colors: [
			"#F7971E",
			"#FFD200"
		]
	},
	{
		name: "Celestial",
		colors: [
			"#C33764",
			"#1D2671"
		]
	},
	{
		name: "Purplepine",
		colors: [
			"#20002c",
			"#cbb4d4"
		]
	},
	{
		name: "Sha la la",
		colors: [
			"#D66D75",
			"#E29587"
		]
	},
	{
		name: "Mini",
		colors: [
			"#30E8BF",
			"#FF8235"
		]
	},
	{
		name: "Maldives",
		colors: [
			"#B2FEFA",
			"#0ED2F7"
		]
	},
	{
		name: "Cinnamint",
		colors: [
			"#4AC29A",
			"#BDFFF3"
		]
	},
	{
		name: "Html",
		colors: [
			"#E44D26",
			"#F16529"
		]
	},
	{
		name: "Coal",
		colors: [
			"#EB5757",
			"#000000"
		]
	},
	{
		name: "Sunkist",
		colors: [
			"#F2994A",
			"#F2C94C"
		]
	},
	{
		name: "Blue Skies",
		colors: [
			"#56CCF2",
			"#2F80ED"
		]
	},
	{
		name: "Chitty Chitty Bang Bang",
		colors: [
			"#007991",
			"#78ffd6"
		]
	},
	{
		name: "Visions of Grandeur",
		colors: [
			"#000046",
			"#1CB5E0"
		]
	},
	{
		name: "Crystal Clear",
		colors: [
			"#159957",
			"#155799"
		]
	},
	{
		name: "Mello",
		colors: [
			"#c0392b",
			"#8e44ad"
		]
	},
	{
		name: "Compare Now",
		colors: [
			"#EF3B36",
			"#FFFFFF"
		]
	},
	{
		name: "Meridian",
		colors: [
			"#283c86",
			"#45a247"
		]
	},
	{
		name: "Relay",
		colors: [
			"#3A1C71",
			"#D76D77",
			"#FFAF7B"
		]
	},
	{
		name: "Alive",
		colors: [
			"#CB356B",
			"#BD3F32"
		]
	},
	{
		name: "Scooter",
		colors: [
			"#36D1DC",
			"#5B86E5"
		]
	},
	{
		name: "Terminal",
		colors: [
			"#000000",
			"#0f9b0f"
		]
	},
	{
		name: "Telegram",
		colors: [
			"#1c92d2",
			"#f2fcfe"
		]
	},
	{
		name: "Crimson Tide",
		colors: [
			"#642B73",
			"#C6426E"
		]
	},
	{
		name: "Socialive",
		colors: [
			"#06beb6",
			"#48b1bf"
		]
	},
	{
		name: "Subu",
		colors: [
			"#0cebeb",
			"#20e3b2",
			"#29ffc6"
		]
	},
	{
		name: "Broken Hearts",
		colors: [
			"#d9a7c7",
			"#fffcdc"
		]
	},
	{
		name: "Kimoby Is The New Blue",
		colors: [
			"#396afc",
			"#2948ff"
		]
	},
	{
		name: "Dull",
		colors: [
			"#C9D6FF",
			"#E2E2E2"
		]
	},
	{
		name: "Purpink",
		colors: [
			"#7F00FF",
			"#E100FF"
		]
	},
	{
		name: "Orange Coral",
		colors: [
			"#ff9966",
			"#ff5e62"
		]
	},
	{
		name: "Summer",
		colors: [
			"#22c1c3",
			"#fdbb2d"
		]
	},
	{
		name: "King Yna",
		colors: [
			"#1a2a6c",
			"#b21f1f",
			"#fdbb2d"
		]
	},
	{
		name: "Velvet Sun",
		colors: [
			"#e1eec3",
			"#f05053"
		]
	},
	{
		name: "Zinc",
		colors: [
			"#ADA996",
			"#F2F2F2",
			"#DBDBDB",
			"#EAEAEA"
		]
	},
	{
		name: "Hydrogen",
		colors: [
			"#667db6",
			"#0082c8",
			"#0082c8",
			"#667db6"
		]
	},
	{
		name: "Argon",
		colors: [
			"#03001e",
			"#7303c0",
			"#ec38bc",
			"#fdeff9"
		]
	},
	{
		name: "Lithium",
		colors: [
			"#6D6027",
			"#D3CBB8"
		]
	},
	{
		name: "Digital Water",
		colors: [
			"#74ebd5",
			"#ACB6E5"
		]
	},
	{
		name: "Orange Fun",
		colors: [
			"#fc4a1a",
			"#f7b733"
		]
	},
	{
		name: "Rainbow Blue",
		colors: [
			"#00F260",
			"#0575E6"
		]
	},
	{
		name: "Pink Flavour",
		colors: [
			"#800080",
			"#ffc0cb"
		]
	},
	{
		name: "Sulphur",
		colors: [
			"#CAC531",
			"#F3F9A7"
		]
	},
	{
		name: "Selenium",
		colors: [
			"#3C3B3F",
			"#605C3C"
		]
	},
	{
		name: "Delicate",
		colors: [
			"#D3CCE3",
			"#E9E4F0"
		]
	},
	{
		name: "Ohhappiness",
		colors: [
			"#00b09b",
			"#96c93d"
		]
	},
	{
		name: "Lawrencium",
		colors: [
			"#0f0c29",
			"#302b63",
			"#24243e"
		]
	},
	{
		name: "Relaxing red",
		colors: [
			"#fffbd5",
			"#b20a2c"
		]
	},
	{
		name: "Taran Tado",
		colors: [
			"#23074d",
			"#cc5333"
		]
	},
	{
		name: "Bighead",
		colors: [
			"#c94b4b",
			"#4b134f"
		]
	},
	{
		name: "Sublime Vivid",
		colors: [
			"#FC466B",
			"#3F5EFB"
		]
	},
	{
		name: "Sublime Light",
		colors: [
			"#FC5C7D",
			"#6A82FB"
		]
	},
	{
		name: "Pun Yeta",
		colors: [
			"#108dc7",
			"#ef8e38"
		]
	},
	{
		name: "Quepal",
		colors: [
			"#11998e",
			"#38ef7d"
		]
	},
	{
		name: "Sand to Blue",
		colors: [
			"#3E5151",
			"#DECBA4"
		]
	},
	{
		name: "Wedding Day Blues",
		colors: [
			"#40E0D0",
			"#FF8C00",
			"#FF0080"
		]
	},
	{
		name: "Shifter",
		colors: [
			"#bc4e9c",
			"#f80759"
		]
	},
	{
		name: "Red Sunset",
		colors: [
			"#355C7D",
			"#6C5B7B",
			"#C06C84"
		]
	},
	{
		name: "Moon Purple",
		colors: [
			"#4e54c8",
			"#8f94fb"
		]
	},
	{
		name: "Pure Lust",
		colors: [
			"#333333",
			"#dd1818"
		]
	},
	{
		name: "Slight Ocean View",
		colors: [
			"#a8c0ff",
			"#3f2b96"
		]
	},
	{
		name: "eXpresso",
		colors: [
			"#ad5389",
			"#3c1053"
		]
	},
	{
		name: "Shifty",
		colors: [
			"#636363",
			"#a2ab58"
		]
	},
	{
		name: "Vanusa",
		colors: [
			"#DA4453",
			"#89216B"
		]
	},
	{
		name: "Evening Night",
		colors: [
			"#005AA7",
			"#FFFDE4"
		]
	},
	{
		name: "Magic",
		colors: [
			"#59C173",
			"#a17fe0",
			"#5D26C1"
		]
	},
	{
		name: "Margo",
		colors: [
			"#FFEFBA",
			"#FFFFFF"
		]
	},
	{
		name: "Blue Raspberry",
		colors: [
			"#00B4DB",
			"#0083B0"
		]
	},
	{
		name: "Citrus Peel",
		colors: [
			"#FDC830",
			"#F37335"
		]
	},
	{
		name: "Sin City Red",
		colors: [
			"#ED213A",
			"#93291E"
		]
	},
	{
		name: "Rastafari",
		colors: [
			"#1E9600",
			"#FFF200",
			"#FF0000"
		]
	},
	{
		name: "Summer Dog",
		colors: [
			"#a8ff78",
			"#78ffd6"
		]
	},
	{
		name: "Wiretap",
		colors: [
			"#8A2387",
			"#E94057",
			"#F27121"
		]
	},
	{
		name: "Burning Orange",
		colors: [
			"#FF416C",
			"#FF4B2B"
		]
	},
	{
		name: "Ultra Voilet",
		colors: [
			"#654ea3",
			"#eaafc8"
		]
	},
	{
		name: "By Design",
		colors: [
			"#009FFF",
			"#ec2F4B"
		]
	},
	{
		name: "Kyoo Tah",
		colors: [
			"#544a7d",
			"#ffd452"
		]
	},
	{
		name: "Kye Meh",
		colors: [
			"#8360c3",
			"#2ebf91"
		]
	},
	{
		name: "Kyoo Pal",
		colors: [
			"#dd3e54",
			"#6be585"
		]
	},
	{
		name: "Metapolis",
		colors: [
			"#659999",
			"#f4791f"
		]
	},
	{
		name: "Flare",
		colors: [
			"#f12711",
			"#f5af19"
		]
	},
	{
		name: "Witching Hour",
		colors: [
			"#c31432",
			"#240b36"
		]
	},
	{
		name: "Azur Lane",
		colors: [
			"#7F7FD5",
			"#86A8E7",
			"#91EAE4"
		]
	},
	{
		name: "Neuromancer",
		colors: [
			"#f953c6",
			"#b91d73"
		]
	},
	{
		name: "Harvey",
		colors: [
			"#1f4037",
			"#99f2c8"
		]
	},
	{
		name: "Amin",
		colors: [
			"#8E2DE2",
			"#4A00E0"
		]
	},
	{
		name: "Memariani",
		colors: [
			"#aa4b6b",
			"#6b6b83",
			"#3b8d99"
		]
	},
	{
		name: "Yoda",
		colors: [
			"#FF0099",
			"#493240"
		]
	},
	{
		name: "Cool Sky",
		colors: [
			"#2980B9",
			"#6DD5FA",
			"#FFFFFF"
		]
	},
	{
		name: "Dark Ocean",
		colors: [
			"#373B44",
			"#4286f4"
		]
	},
	{
		name: "Evening Sunshine",
		colors: [
			"#b92b27",
			"#1565C0"
		]
	},
	{
		name: "JShine",
		colors: [
			"#12c2e9",
			"#c471ed",
			"#f64f59"
		]
	},
	{
		name: "Moonlit Asteroid",
		colors: [
			"#0F2027",
			"#203A43",
			"#2C5364"
		]
	},
	{
		name: "MegaTron",
		colors: [
			"#C6FFDD",
			"#FBD786",
			"#f7797d"
		]
	},
	{
		name: "Cool Blues",
		colors: [
			"#2193b0",
			"#6dd5ed"
		]
	},
	{
		name: "Piggy Pink",
		colors: [
			"#ee9ca7",
			"#ffdde1"
		]
	},
	{
		name: "Grade Grey",
		colors: [
			"#bdc3c7",
			"#2c3e50"
		]
	}
];

function _createForOfIteratorHelper(o, allowArrayLike) { var it; if (typeof Symbol === "undefined" || o[Symbol.iterator] == null) { if (Array.isArray(o) || (it = _unsupportedIterableToArray$1(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = o[Symbol.iterator](); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it.return != null) it.return(); } finally { if (didErr) throw err; } } }; }

function _unsupportedIterableToArray$1(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray$1(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray$1(o, minLen); }

function _arrayLikeToArray$1(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

var debounce = function debounce(fn) {
  var ms = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
  var timeoutId;
  return function () {
    var _this = this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    clearTimeout(timeoutId);
    timeoutId = setTimeout(function () {
      return fn.apply(_this, args);
    }, ms);
  };
};

var toCamelCase = function toCamelCase(str) {
  var s = str && str.match(/[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g).map(function (x) {
    return x.slice(0, 1).toUpperCase() + x.slice(1).toLowerCase();
  }).join('');
  return s.slice(0, 1).toLowerCase() + s.slice(1);
};

var uigradientsData = uigradients.reduce(function (rs, item) {
  rs[toCamelCase(item.name)] = item.colors;
  return rs;
}, {});
var exportModule = window || {};
var methodName = 'initCMUI';

var checkDevice = function checkDevice() {
  return /iphone|ipad|android|micromessenger/i.test(window.navigator.appVersion) || document.scrollingElement.clientWidth < 768;
};

var isMobile = checkDevice();
var isInit = false;
var styleInit = (function () {
  if (!window || !document) {
    return function () {};
  }

  function setFontSize() {
    if (document.body) {
      document.documentElement.style.fontSize = isMobile ? '10vw' : '75px';
      document.body.style.fontSize = '14px';
    } else {
      document.addEventListener('DOMContentLoaded', setFontSize);
    }
  }

  setFontSize(); // insert style

  var cmuiStyle = document.createElement('style');
  document.head.appendChild(cmuiStyle);
  var cssRules = cmuiStyle.sheet.cssRules;
  var cssRulesLen = cssRules.length;

  function insertStyle(selector, name, value) {
    name = name.replace(/[A-Z]/g, function (i) {
      return "-".concat(i.toLowerCase());
    });
    cmuiStyle.sheet.insertRule("".concat(selector, "{").concat(name, ":").concat(value, "}"), cssRulesLen);
  } // set gradient


  function setGradient() {
    var index = 1;

    var _loop = function _loop(key) {
      ['', 'top', 'bottom', 'left'].forEach(function (pos) {
        insertStyle(".gradient_".concat(key).concat(pos ? '_' + pos : '', ",.gradient_").concat(index).concat(pos ? '_' + pos : ''), "background", "linear-gradient(to ".concat(pos || 'right', ", ").concat(uigradientsData[key].join(','), ")"));
      });
      index++;
    };

    for (var key in uigradientsData) {
      _loop(key);
    }
  }

  setGradient(); // set pm

  var PMstore = new Set();
  var reg = /\b((padding|margin|radius)[trblvh]?|top|bottom|right|left)(-n)?\d+\b/g;
  var nameObject = {
    t: ['Top'],
    l: ['Left'],
    r: ['Right'],
    b: ['Bottom'],
    v: ['Top', 'Bottom'],
    h: ['Left', 'Right']
  };

  function setPMBySet(set) {
    var _iterator = _createForOfIteratorHelper(set),
        _step;

    try {
      var _loop2 = function _loop2() {
        var item = _step.value;

        // ratio
        if (~item.indexOf('ratio')) {
          var _$exec = /ratio(\d+)\/(\d+)/.exec(item),
              _$exec2 = _slicedToArray(_$exec, 3),
              x = _$exec2[1],
              y = _$exec2[2];

          insertStyle(".ratio-container[ratio=\"".concat(x, "/").concat(y, "\"]::before"), 'padding-top', "".concat(y / x * 100, "%"));
          return "continue";
        }

        var _ref = /(padding|margin|radius)([trblvh])?(-n)?(\d+)/.exec(item) || /(top|left|right|bottom)()?(-n)?(\d+)/.exec(item),
            _ref2 = _slicedToArray(_ref, 5),
            key = _ref2[0],
            name = _ref2[1],
            pos = _ref2[2],
            isN = _ref2[3],
            value = _ref2[4]; // radius


        if (name === 'radius') {
          [[".".concat(key), 'border-radius', "".concat(value, "px")], [".".concat(key, ".list::before, .").concat(key, ".border.light.item > *"), 'border-radius', "".concat(value * 2, "px")], [".inputGroup input.".concat(key, ":first-child, .btn-group .btn.").concat(key, ":first-child, .badge-group .badge.").concat(key, ":first-child"), 'border-radius', "".concat(value, "px")], [".inputGroup input.".concat(key, ":last-child, .btn-group .btn.").concat(key, ":last-child, .badge-group .badge.").concat(key, ":last-child"), 'border-radius', "".concat(value, "px")]].forEach(function (item) {
            insertStyle.apply(void 0, _toConsumableArray(item));
          });
          return "continue";
        }

        value = isMobile ? "".concat(value / 75, "rem") : "".concat(value, "px");

        if (pos) {
          nameObject[pos].forEach(function (posName) {
            insertStyle('.' + key, "".concat(name).concat(posName), (isN ? '-' : '') + value);
          });
        } else {
          insertStyle('.' + key, "".concat(name), (isN ? '-' : '') + value);
        }
      };

      for (_iterator.s(); !(_step = _iterator.n()).done;) {
        var _ret = _loop2();

        if (_ret === "continue") continue;
      }
    } catch (err) {
      _iterator.e(err);
    } finally {
      _iterator.f();
    }
  }

  function setPMByClass(className) {
    var PMstoreTemp = new Set();
    var match = (_toConsumableArray(className).join(' ').match(reg) || []).concat(_toConsumableArray(className).join(' ').match(/ratio\d+\/\d+/g) || []);
    match.forEach(function (item) {
      if (!PMstore.has(item)) {
        PMstore.add(item);
        PMstoreTemp.add(item);
      }
    });
    setPMBySet(PMstoreTemp);
  }

  function setPMByDom() {
    var dom = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : document;
    var PMstoreTemp = new Set();
    var domList = dom.querySelectorAll('[class*=padding],[class*=margin],[class*=top],[class*=right],[class*=left],[class*=bottom],[class*=radius],.ratio-container[ratio]') || [];

    _toConsumableArray(domList).forEach(function (item) {
      // ratio
      if (item.classList.contains('ratio-container') && /\d+\/\d+/.test(item.getAttribute('ratio'))) {
        var ratio = item.getAttribute('ratio');
        ratio = "ratio".concat(ratio);

        if (!PMstore.has(ratio)) {
          PMstore.add(ratio);
          PMstoreTemp.add(ratio);
        }

        return;
      }

      var match = item.className.match(reg) || [];
      match.forEach(function (item) {
        if (!PMstore.has(item)) {
          PMstore.add(item);
          PMstoreTemp.add(item);
        }
      });
    });

    setPMBySet(PMstoreTemp);
  }

  function obs() {
    !isInit && new MutationObserver(function (mutations) {
      isInit = true;
      var temp = new Set();
      var classList = new Set();
      mutations.reduce(function (rs, item) {
        if (item.type === 'attributes') {
          switch (item.attributeName) {
            case 'class':
              _toConsumableArray(item.target.classList).forEach(function (className) {
                return classList.add(className);
              });

              break;

            case 'ratio':
              classList.add('ratio' + item.target.getAttribute('ratio'));
          }
        } else if (item.type === 'childList' && item.addedNodes.length) {
          temp.add(item.target);
        }
      }, {});

      _toConsumableArray(temp).forEach(setPMByDom);

      classList.size && setPMByClass(classList);
    }).observe(document.body, {
      childList: true,
      subtree: true,
      attributeFilter: ['class', 'ratio']
    });
  }
  /* eslint-enable */
  // set initCMUI


  function addMethod(name, fn) {
    var oldObject = exportModule[name];

    exportModule[name] = function () {
      if (fn.length === arguments.length) return fn.apply(this, arguments);

      if (typeof oldObject === 'function') {
        return oldObject.apply(this, arguments);
      }
    };
  }

  addMethod(methodName, initByAuto);
  addMethod(methodName, initByNumber);
  addMethod(methodName, initByRange);
  var resetRule = debounce(function () {
    if (checkDevice() !== isMobile) {
      isMobile = checkDevice();
      setFontSize();
      Array.prototype.slice.call(cmuiStyle.sheet.rules).forEach(function (rule) {
        var style = rule.style;
        var len = style.length;

        while (len--) {
          var name = style[len].replace(/-(\w)/g, function (word, $1) {
            return $1.toUpperCase();
          });
          var matchs = style[name].match(/\d+(\.\d+)?/g);

          if (matchs) {
            var value = matchs[0];

            if (isMobile) {
              style[name] = value / 75 + 'rem';
            } else {
              style[name] = value * 75 + 'px';
            }
          }
        }
      });
    }
  }, 500);
  window.addEventListener('resize', resetRule, false);

  function initByAuto() {
    if (document.body) {
      obs();
      setPMByDom();
    } else {
      document.addEventListener('DOMContentLoaded', initByAuto, false);
    }
  }

  function setStep(number) {
    var step = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 1;
    var data = new Set();

    var _loop3 = function _loop3(i) {
      ['padding', 'margin'].forEach(function (name) {
        ['t', 'r', 'b', 'l', 'h', 'v'].forEach(function (pos) {
          data.add("".concat(name).concat(pos).concat(i));
        });
        data.add("".concat(name).concat(i));
      });
    };

    for (var i = 0; i < number; i += step) {
      _loop3(i);
    }

    setPMBySet(data);
  }

  function initByNumber(number) {
    setStep(number);
  }

  function initByRange(number, step) {
    setStep(number, step);
  }

  return exportModule[methodName];
})();

export default styleInit;
